<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Metagenomics - ABRPI Training 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics Workshop <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../workshop_overview/">Workshop overview</a>
</li>

                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li class="active">
    <a href="./">Metagenomics</a>
</li>

                    
                        
<li >
    <a href="../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                        
<li >
    <a href="../kallisto/">Differential gene expression using Kallisto and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xtandem/">Protein Identification</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Pathway Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../pathway_tools/annotation/">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../workflows/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../pacbio/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#metagenomics">Metagenomics</a></li>
        
            <li><a href="#get-data">Get data</a></li>
        
            <li><a href="#1-pre-process">1. Pre-process</a></li>
        
            <li><a href="#2-clustering-swarm">2. Clustering swarm</a></li>
        
            <li><a href="#3-remove-chimera">3. Remove chimera</a></li>
        
            <li><a href="#4-filters">4. Filters</a></li>
        
            <li><a href="#5-affiliation-otu">5. Affiliation OTU</a></li>
        
            <li><a href="#6-affiliations-stat">6. Affiliations stat</a></li>
        
            <li><a href="#links">Links</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="metagenomics">Metagenomics</h1>
<p>Metagenomics aims to compare microbial communities from different environments by using information from the metagenome. Typically, 16S rRNA is used when classifying taxa, and whole-genome sequencing when aiming to identify gene functions and pathways.</p>
<!-- - compare microbial communities from different environments: do this by comparing the metagenome: e.g. GC-content, k-mers, size, how many spp, functions of genes
- eg for function, can compare to a database (COG or KEGG) and see whether metagenomes differ (statistically test) -
- More about 16S rRNA: The 16S sequence is RNA that makes up part of the bacterial ribosome (together with other rRNA and protein). The operon is 16S, 23S, 5S - and there may be many copies of the operon (7 in E coli).
-->

<p>This tutorial covers the tool called FROGS (in Galaxy): &ldquo;Find Rapidly OTU with Galaxy Solution&rdquo;.
<!-- dental and phln only? --></p>
<!-- - uses a different way of clustering similar sequences into OTUs. -->

<h2 id="get-data">Get data</h2>
<ul>
<li>The data: paired-end Illumina reads from two environmental samples. <!-- two env samples? dental microbiomes?--></li>
</ul>
<!-- uplod not working
- In Galaxy, in the history panel, click on the cog item, and select "import from file". In the "Archived History URL:" box paste: https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/16SmetaG_dataset.tar.gz

-->

<ul>
<li>Download the <fn>.tar.gz</fn> file to your local computer using this link: <a href="https://drive.google.com/open?id=0B-DKulJTTftlT2djQmJ4aTN6Q1U">https://drive.google.com/open?id=0B-DKulJTTftlT2djQmJ4aTN6Q1U</a>.</li>
<li>Click on the file to unzip it into four <fn>.gz</fn> files. These can stay zipped.</li>
<li>In Galaxy, go to <ss>Get Data: Upload File</ss>. Choose the the four files, set their file type to &ldquo;fastqsanger&rdquo;, click start, then close.</li>
<li>The four files should now be in your current history.</li>
</ul>
<!-- would be clearer to rename these files as something like microbiomeA_R1 and microbiomeB_R1 etc -->

<ul>
<li>In the Galaxy tools panel, in the top search bar, type FROGS.</li>
<li>This will bring up the various tools available.</li>
<li>We will be using 6 of these tools.</li>
</ul>
<p><img alt="frogs_tools" src="./images/frogs_tools.png" /></p>
<h2 id="1-pre-process">1. Pre-process</h2>
<p>This is the first step in the Frogs analysis. It takes in the forward and reverse sequencing reads (R1 and R2) from multiple samples (e.g. microbiome A and microbiome B) and performs the following steps:</p>
<ul>
<li>If the data is not in contigs, read1 and read2 will be overlapped (allowing some mismatch in overlapping region).</li>
<li>Contigs that are too big or too small will be filtered out. <!-- what's wrong with too big?--></li>
<li>If using the Illumina standard protocol: will look for those primers, filter out contigs without primers, and cut the primer sequences.</li>
<li>Sequences that are too small or of poor quality will be filtered out.</li>
<li>Sequences will be de-replicated: duplicates will be removed but the number of duplicates will be recorded.</li>
</ul>
<h3 id="run">Run</h3>
<p>Go to <ss>FROGS Pre-process</ss> and select the following:</p>
<ul>
<li><ss>Sequencer</ss>: Illumina</li>
<li><ss>Input type</ss>: Files by samples</li>
<li><ss>Reads alread contiged?</ss>: No</li>
<li><ss>Samples</ss>:<ul>
<li><ss>Name</ss>: microbiomeA &lt;<em>note: no spaces</em>&gt;</li>
<li><ss>Reads1</ss>: <fn>microbiomeA_R1.fq</fn></li>
<li><ss>Reads2</ss>: <fn>microbiomeA_R2.fq</fn></li>
<li><ss>Insert Samples</ss>- click plus sign</li>
<li><ss>Name</ss>: microbiomeB</li>
<li><ss>Reads1</ss>: <fn>microbiomeB_R1.fq</fn></li>
<li><ss>Reads2</ss>: <fn>microbiomeB_R2.fq</fn></li>
</ul>
</li>
<li><ss>Reads 1 size</ss>: 250</li>
<li><ss>Reads 2 size</ss>: 250</li>
<li><ss>Expected amplicon size</ss>: 420</li>
<li><ss>Minimum amplicon size</ss>: 380</li>
<li><ss>Maximum amplicon size</ss>: 460</li>
<li><ss>Sequencing protocol</ss>: Illumina standard</li>
<li><ss>5&rsquo; primer</ss>: GGCGVACGGGTGAGTAA</li>
<li><ss>3&rsquo; primer</ss>: GTGCCAGCNGCNGCGG &lt;<em>note: needs to be in 5&rsquo; to 3&rsquo; orientation</em>&gt;</li>
<li><ss>Execute</ss></li>
</ul>
<p><img alt="pre-process" src="./images/pre-process.png" /></p>
<h3 id="output">Output</h3>
<p>There are three output files. Click on the eye icon for each to see their contents.</p>
<p>Report:</p>
<ul>
<li>The <fn>report.html</fn> shows how samples were filtered. For example, in the bar chart called &ldquo;Filtering summary&rdquo;, the number of reads kept at each filtering stage is displayed. We would expect some decrease in some of these categories, but if many reads have been filtered at a particular stage it could indicate poor quality data.</li>
</ul>
<p><img alt="filtering_summary1" src="./images/filtering_summary1.png" /></p>
<ul>
<li>Below this chart, a table shows the statistics for each sample. Check the boxes next to &ldquo;microbiomeA&rdquo; and &ldquo;microbiomeB&rdquo; and then click <ss>Display amplicon lengths</ss> - shows amplicon distribution before filtering. Click <ss>Display preprocessed amplicon lengths</ss> - shows the distribution has been narrowed after filtering. Check if both samples are similar.</li>
</ul>
<p>Counts:</p>
<ul>
<li>The <fn>counts.tsv</fn> file is a list of sequences and their counts in the two samples.</li>
</ul>
<p>Fasta file:</p>
<ul>
<li>The <fn>dereplicated.fasta</fn> contains the sequences, without any duplicates.</li>
</ul>
<h2 id="2-clustering-swarm">2. Clustering swarm</h2>
<p>In this step, sequences are clustered into groups using Swarm (more information about <a href="https://peerj.com/articles/593/">Swarm</a>). This takes the pre-processed <fn>fasta</fn> and <fn>counts</fn> files and does the following:</p>
<ul>
<li>Sorts reads by abundance.</li>
<li>Clusters the reads into pre-clusters using Swarm and distance parameter of 1.</li>
<li>Sorts these pre-clusters by abundance.</li>
<li>Cluster the pre-clusters using Swarm and a user-specified distance.</li>
</ul>
<p>Go to <ss>FROGS Clustering swarm</ss> and select the following:</p>
<ul>
<li><ss>Sequences files</ss>: <fn>dereplicated.fasta</fn></li>
<li><ss>Counts file</ss>: <fn>count.tsv</fn></li>
<li><ss>Aggregation distance</ss>: 3</li>
<li><ss>Perform deionising clustering step?</ss>: Yes</li>
<li><ss>Execute</ss></li>
</ul>
<p><em>Note: Galaxy may say that the job submission has failed. Click Close, then click refresh in the top of the history panel. The job should be running.</em></p>
<p><img alt="clustering" src="./images/clustering.png" /></p>
<h3 id="output_1">Output</h3>
<p>There are three output files.</p>
<ul>
<li>The <fn>abundance.biom</fn> shows the abundance of each cluster.</li>
<li>The <fn>seed_sequences.fasta</fn> contains the the cluster (OTU) representative sequences. <!-- the seqs used to start clustering? not the final clusterss? --></li>
<li>The <fn>swarms.composition.tsv</fn> shows what is in each cluster.</li>
</ul>
<h2 id="3-remove-chimera">3. Remove chimera</h2>
<p>Closely-related sequences may form chimeras (mixed sequences) during PCR (libray prep). This step removes these sequences by the following method:</p>
<ul>
<li>Splits input data into samples</li>
<li>Uses vsearch to find chimeras in each sample</li>
<li>Removes chimeras</li>
</ul>
<p>Go to <ss>FROGS Remove chimera</ss> and select the following:</p>
<ul>
<li><ss>Sequences file</ss>: <fn>seed_sequences.fasta</fn></li>
<li><ss>Abundance type</ss>: <fn>BIOM file</fn></li>
<li><ss>Abundance file</ss>: <fn>abundance.biom</fn></li>
<li><ss>Execute</ss></li>
</ul>
<p><em>Note: Galaxy may say that the job submission has failed. Click Close, then click refresh in the top of the history panel. The job should be running.</em></p>
<p><img alt="chimera" src="./images/chimera.png" /></p>
<h3 id="output_2">Output</h3>
<p>There are three output files.</p>
<ul>
<li>The <fn>non_chimera.fasta</fn> is a filtered file containing no chimeras.</li>
<li>The <fn>non_chimera_abundance.biom</fn> is a filtered abundance file containing no chimeras.</li>
<li>The summary <fn>report.html</fn>. In this case, although almost 70% of the clusters were removed, more than 70% of the actual reads were kept.</li>
</ul>
<p><img alt="remove-summary" src="./images/remove_summary.png" /></p>
<h2 id="4-filters">4. Filters</h2>
<p>The OTUs (Operational Taxonomic Units) have now been clustered. In this step, we will filter out some of the OTUs that are either not in both samples, and/or contain at least 0.005% of all the sequences.</p>
<p>Go to <ss>FROGS Filters</ss> and select the following:</p>
<ul>
<li><ss>Sequences file</ss>: <fn>non_chimera.fasta</fn></li>
<li><ss>Abundance file</ss>:  <fn>non_chimera_abundance.biom</fn></li>
<li><ss>*** THE FILTERS ON OTUS IN SAMPLES, OTUS SIZE and SEQUENCE PERCENTAGE</ss>: Apply filters</li>
<li><ss>Minimum number of samples</ss>: 2</li>
<li><ss>Minimum proportion/number of sequences to keep OTU</ss>: 0.00005</li>
<li><ss>N biggest OTU</ss>: leave blank</li>
<li><ss>*** THE FILTERS ON RDP</ss>: No filters</li>
<li><ss>*** THE FILTERS ON BLAST</ss>: No filters</li>
<li><ss>*** THE FILTERS ON CONTAMINATIONS</ss>: No filters</li>
<li><ss>Execute</ss></li>
</ul>
<p><img alt="filters" src="./images/filters.png" />]</p>
<p><em>Note: Galaxy may say that the job submission has failed. Click Close, then click refresh in the top of the history panel. The job should be running.</em></p>
<h3 id="output_3">Output</h3>
<p>There are four output files.</p>
<ul>
<li><fn>sequences.fasta</fn>: the retained sequences.</li>
<li><fn>abudance.biom</fn>: the abundance file.</li>
<li><fn>excluded.tsv</fn>: discarded OTUs</li>
<li><fn>report.html</fn>: this shows the proportion of OTUs and the proportion of sequences kept. In this example, although most OTUs have been filtered out, most sequences have been retained.</li>
</ul>
<p><img alt="filters_summary" src="./images/filters_summary.png" />
<!-- We should do 3+ samples
Then could click on Venn diagram here and see intersection.
--></p>
<h2 id="5-affiliation-otu">5. Affiliation OTU</h2>
<p>An OTU is a cluster of sequences. This step adds the taxonomy to the abundance file. It uses the SILVA database for rRNA.<!--add more info about Silva--></p>
<!-- - that has been filtered for only (16S or all 30S?), non-redundant, keeping certain taxonomic levels, and split into pro and eukaryotes.

blastn+ to align each OTU against seqs in the dbase, keeping the best.
- It can return multi-affiliation - see notes below tool panel.

-->

<p>Go to <ss>FROGS Affiliation OTU</ss> and select:</p>
<ul>
<li><ss>Using reference database</ss>: silva123</li>
<li><ss>Also perform RDP assignation</ss>: No</li>
<li><ss>OTU seed sequence</ss>: <fn>sequences.fasta</fn> from step 4</li>
<li><ss>Abundance file</ss>: <fn>abundance.biom</fn> from step 4</li>
<li><ss>Execute</ss></li>
</ul>
<p><img alt="otu" src="./images/otu.png" /></p>
<p><em>Note: Galaxy may say that the job submission has failed. Click Close, then click refresh in the top of the history panel. The job should be running.</em></p>
<!-- can take 10 mins or so? -->

<h3 id="output_4">Output</h3>
<p>There are two output files.</p>
<ul>
<li>
<p><fn>affiliation.biom</fn>: the abundance file with affiliation. Note: this <fn>.biom</fn> file is not human-readable. If you wish, you can convert it with the FROGS <strong>biom to tsv</strong> tool.</p>
</li>
<li>
<p><fn>report.html</fn>: the report shows the proportion of OTUs and sequences affiliated to sequences in the database. Here we can see that almost all OTUs and sequences have been taxonomically assigned, but almost 10% of OTUs could be assigned to multiple species. <!--is that interpreation correct?--></p>
</li>
</ul>
<p><img alt="blast_summary" src="./images/blast_affiliation.png" /></p>
<h2 id="6-affiliations-stat">6. Affiliations stat</h2>
<p>This step computes some statistics from the analysis and generates a report of the OTUs/taxonomy found.</p>
<p>Go to <ss>FROGS Affiliations stat</ss> and select:</p>
<ul>
<li><ss>Abundance file</ss>: <fn>affiliation.biom</fn> from step 5</li>
<li><ss>Rarefaction ranks</ss>: Class Order Family Genus Species</li>
<li><ss>Affiliation processed</ss>: FROGS blast</li>
<li><ss>Execute</ss></li>
</ul>
<p><img alt="otu-stat" src="./images/otu-stat.png" /></p>
<h3 id="output_5">Output</h3>
<p>There is one output file. Click on the eye icon next to <fn>summary.html</fn>:</p>
<ul>
<li>click on <ss>Display global distribution</ss>: this shows the taxonomy of the sample. Start at the centre; major groups each have a segment by colour; as you go outwards, taxonomic identification becomes more specific. <!-- this is the pooled info from both samples? --> Click on a segment to display a table of statisics below. Click on the cross in the top right corner to exit.</li>
</ul>
<p><img alt="taxa-distribution" src="./images/taxa_distribution.png" /></p>
<!--is there a way to summarize this info across samples? eg sample 1 had lots of taxon A compared to sample 2 etc... or this taxon found only in sample B etc... (although I think these ones were filtered out)? -->

<ul>
<li>Next, we will look at the rarefaction curve, which is a measure of samples vs diversity. <!-- is that correct? In other words, how many samples do we have to take before we are covering the taxonomic diversity?--> Click on the boxes next to each of the samples (both, or one at a time) in the table (microbiomeA and microbiomeB). Click <ss>Display rarefaction</ss>.</li>
</ul>
<p><img alt="rarefaction" src="./images/rarefaction.png" /></p>
<h2 id="links">Links</h2>
<ul>
<li>FROGS slides by Yvan Le Bras <a href="https://f1000research.com/slides/5-1832">https://f1000research.com/slides/5-1832</a></li>
<li>The SILVA database: <a href="https://www.arb-silva.de/">https://www.arb-silva.de/</a></li>
<li>Other metagenomics software:<ul>
<li>QIIME: <a href="http://qiime.org/">http://qiime.org/</a>    </li>
<li>UPARSE: <a href="http://www.drive5.com/uparse/">http://www.drive5.com/uparse/</a></li>
<li>MOTHUR: <a href="https://www.mothur.org/">https://www.mothur.org/</a></li>
<li>MG-RAST: <a href="http://metagenomics.anl.gov/">http://metagenomics.anl.gov/</a></li>
</ul>
</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>