
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
        <link rel="canonical" href="http://sepsis-omics.github.io/tutorials/modules/pacbio/">
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <link rel="shortcut icon" href="../../favicon.ico">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.6.1">
    
    
      
        <title>PacBio assembly with SMRT portal - ABRPI</title>
      
    
    
      <script src="../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application-4d0d3f2fbf.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../css/sepsis.css">
    
    
  </head>
  
  
  
  
    <body>
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="http://sepsis-omics.github.io/tutorials/" title="ABRPI" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Genomics
                </span>
              
            
            PacBio assembly with SMRT portal
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
            


  


  <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../workshop_overview/" title="Genomics" class="md-tabs__link md-tabs__link--active">
          Genomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../dge/" title="Transcriptomics" class="md-tabs__link">
          Transcriptomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xtandem/" title="Proteomics" class="md-tabs__link">
          Proteomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../xcms/" title="Metabolomics" class="md-tabs__link">
          Metabolomics
        </a>
      
    </li>
  

      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    ABRPI
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2" checked>
    
    <label class="md-nav__link" for="nav-2">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../workshop_overview/" title="Genomics workshop overview" class="md-nav__link">
      Genomics workshop overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pear/" title="Merge reads with Pear" class="md-nav__link">
      Merge reads with Pear
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../velvet/" title="Genome assembly with Velvet" class="md-nav__link">
      Genome assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades/" title="Genome assembly with Spades" class="md-nav__link">
      Genome assembly with Spades
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../snippy/" title="Variant finding" class="md-nav__link">
      Variant finding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../access_data/" title="Public data &rarr; assembly, annotation" class="md-nav__link">
      Public data &rarr; assembly, annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../abricate/" title="Finding antibiotic-resistant genes" class="md-nav__link">
      Finding antibiotic-resistant genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../viral_genomes/" title="Viral genome sequencing" class="md-nav__link">
      Viral genome sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../workflows/" title="Galaxy workflows" class="md-nav__link">
      Galaxy workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../frogs/" title="Metagenomics with FROGS" class="md-nav__link">
      Metagenomics with FROGS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kraken/" title="Sample identification with Kraken" class="md-nav__link">
      Sample identification with Kraken
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        PacBio assembly with SMRT portal
      </label>
    
    <a href="./" title="PacBio assembly with SMRT portal" class="md-nav__link md-nav__link--active">
      PacBio assembly with SMRT portal
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#start" title="Start" class="md-nav__link">
    Start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input" title="Input" class="md-nav__link">
    Input
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" title="Assembly" class="md-nav__link">
    Assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hgap-process-overview" title="HGAP process overview" class="md-nav__link">
    HGAP process overview
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#set-up-job" title="Set up job" class="md-nav__link">
    Set up job
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs-and-outputs" title="Inputs and Outputs" class="md-nav__link">
    Inputs and Outputs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" title="Results" class="md-nav__link">
    Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output" title="Output" class="md-nav__link">
    Output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#links-to-more-information" title="Links to more information" class="md-nav__link">
    Links to more information
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../cmdline_assembly/" title="PacBio assembly with command line tools" class="md-nav__link">
      PacBio assembly with command line tools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../spades_cmdline/" title="Spades assembly - commandline" class="md-nav__link">
      Spades assembly - commandline
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../mauve/" title="Genome alignment with Mauve" class="md-nav__link">
      Genome alignment with Mauve
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary/" title="Pangenomes with Roary/Phandango - command line" class="md-nav__link">
      Pangenomes with Roary/Phandango - command line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../roary-galaxy/" title="Pangenomes with Roary/Phandango - Galaxy" class="md-nav__link">
      Pangenomes with Roary/Phandango - Galaxy
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../dge/" title="Differential gene expression using Galaxy and Degust" class="md-nav__link">
      Differential gene expression using Galaxy and Degust
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../kallisto/" title="Differential gene expression using Kallisto and Degust" class="md-nav__link">
      Differential gene expression using Kallisto and Degust
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xtandem/" title="Protein Identification" class="md-nav__link">
      Protein Identification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../pathway_tools/annotation/" title="Pathway annotation and prediction" class="md-nav__link">
      Pathway annotation and prediction
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Metabolomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metabolomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../xcms/" title="Metabolite identification" class="md-nav__link">
      Metabolite identification
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#start" title="Start" class="md-nav__link">
    Start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#input" title="Input" class="md-nav__link">
    Input
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" title="Assembly" class="md-nav__link">
    Assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#hgap-process-overview" title="HGAP process overview" class="md-nav__link">
    HGAP process overview
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#set-up-job" title="Set up job" class="md-nav__link">
    Set up job
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inputs-and-outputs" title="Inputs and Outputs" class="md-nav__link">
    Inputs and Outputs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#results" title="Results" class="md-nav__link">
    Results
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#output" title="Output" class="md-nav__link">
    Output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#next" title="Next" class="md-nav__link">
    Next
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#links-to-more-information" title="Links to more information" class="md-nav__link">
    Links to more information
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <p><br></p>
<h1 id="assembly-with-pacbio-data-and-smrt-portal">Assembly with PacBio data and SMRT Portal</h1>
<p>Keywords: de novo assembly, PacBio, PacificBiosciences, HGAP, SMRT Portal, Microbial Genomics Virtual Laboratory</p>
<p>This tutorial will show you how to assemble a bacterial genome <em>de novo</em>, using the PacBio SMRT Portal on the mGVL. We will use an analysis pipeline called HGAP, the Hierarchical Genome Assembly Process.</p>
<!-- FIXMEs:

- will SMRT portal be available on all training GVLs
- will students use existing SMRT portal registrations, or will each set up their own
- PacBio data training data to be loaded into SMRT portal(s) -->

<!-- [How to get data => GVL] -->

<h2 id="start">Start</h2>
<ul>
<li>Open your mGVL dashboard.</li>
<li>You should see SMRT Portal as one of the instance services on your GVL dashboard.</li>
<li>Open up the SMRT portal web link (to the right) and register/log on.</li>
</ul>
<h2 id="input">Input</h2>
<!-- We will use a publicly available data-set of PacBio reads from the bacteria *E. coli* (reference link below).
-->

<p>We will use a dataset from a <em>Streptococcus pyogenes</em> bacteria.</p>
<p>If this has already been loaded onto SMRT portal (e.g. for use during a workshop), proceed to the next step (&ldquo;Assembly&rdquo;).</p>
<p>Otherwise:</p>
<ul>
<li>Load the PacBio data (your own, or the training dataset) onto your GVL.</li>
<li>In the SMRT Portal, go to <ss>Design Job</ss>, the top left tab.</li>
<li>Go to <ss>Import and Manage</ss>.
<img alt="smrt portal screenshot" src="./images/image03.png" /></li>
<li>Click <ss>Import SMRT cells</ss>.
<img alt="smrt portal screenshot" src="./images/image04.png" /></li>
<li>
<p>Work out where you put the data on your GVL, and make sure the file path is showing.</p>
<ul>
<li>If not, click <ss>Add</ss> and enter the file path to the data.</li>
<li>A SMRT cell is the collection of data from a particular cell in the machine. It includes .bax.h5 files.</li>
</ul>
</li>
<li>
<p>Click on the file path and then <ss>Scan</ss> to check for new data.</p>
</li>
</ul>
<h2 id="assembly">Assembly</h2>
<h3 id="hgap-process-overview">HGAP process overview</h3>
<p>We will use the Hierarchical Genome Assembly Process (HGAP). This flowchart shows the steps in the process:</p>
<p><img alt="flowchart of HGAP process" src="./images/flowchart.png" /></p>
<h3 id="set-up-job">Set up job</h3>
<ul>
<li>In the SMRT Portal, go to the top left tab, <ss>Design Job</ss>.</li>
<li>Go to <ss>Create New</ss>.</li>
<li>An <ss>Analysis</ss> window should appear. Tick all the boxes, then <ss>Next</ss>.</li>
<li>Under <ss>Job Name</ss> enter a name.</li>
<li>To the right, under <ss>Groups</ss> choose <em>all</em>.</li>
<li>Under <ss>Protocols</ss> choose <ss>RS_HGAP_Assembly.3</ss>.</li>
<li>There is an ellipsis underneath <ss>Protocols</ss> - click on the ellipsis.</li>
</ul>
<p><img alt="smrt portal screenshot" src="./images/setup.png" /></p>
<p>This brings up the settings. Click on <ss>Assembly</ss>.</p>
<ul>
<li>
<p>For <ss>Compute Minimum Seed Read Length</ss>: ensure box is ticked
<!--
Sarah Baines:
("compute minimum seed read length"). While this protocol works ok, I have found that the longer the average seed read length (larger library cuts), the estimation protocol can get it really wrong, and frequently if the cell has low output this will cause Celera to fail very early on. Performing this estimation manually can be difficult (as you know!) but it may be worth adding a comment or something.
--></p>
</li>
<li>
<p>For <ss>Number of Seed Read Chunks</ss>: enter <em>12</em></p>
</li>
<li>Change the <ss>Genome Size</ss> to an approximately correct size for the species. For <em>S. pyogenes</em>, enter 1800000.</li>
<li>For <ss>Target Coverage</ss>: enter <em>10</em></li>
<li>For <ss>Overlapper Error Rate</ss>: enter <em>0.04</em></li>
<li>Leave all other settings as they are.</li>
<li>Click <ss>Apply</ss></li>
</ul>
<p>Your protocol window should look like this:</p>
<p><img alt="smrt portal screenshot" src="./images/assembly_options.png" /></p>
<ul>
<li>
<p>Click <ss>Ok</ss>.  </p>
</li>
<li>
<p>In the <ss>SMRT Cells Available</ss> window, select the file to be used. Click on the arrow to transfer these files to the SMRT Cells in Job window.</p>
</li>
<li>You can drag the column widths of the &ldquo;Url&rdquo; column so that you can see the URLs of the file paths better.</li>
</ul>
<p><img alt="smrt portal screenshot" src="./images/smrt3.png" /></p>
<ul>
<li>Click <ss>Save</ss> (bottom right hand side).</li>
<li>Next to <ss>Save</ss>, click <ss>Start</ss>.</li>
<li>The <ss>Monitor Jobs</ss> window should open.<ul>
<li>As each step proceeds, new items will appear under the <ss>Reports</ss> and <ss>Data</ss> tabs on the left.</li>
</ul>
</li>
</ul>
<p><img alt="smrt portal screenshot" src="./images/smrt6.png" /></p>
<h3 id="inputs-and-outputs">Inputs and Outputs</h3>
<p>The connections between the names of assembly stages and outputs is not always clear. This flowchart shows how each stage of the HGAP process corresponds to protocol window names and outputs:</p>
<p><img alt="inputs and outputs" src="./images/inputs_outputs.png" />
<!-- upload a better res image? --></p>
<h2 id="results">Results</h2>
<p>If the job is still running, click on the centre tab <ss>Monitor Jobs</ss>. Otherwise, click on the top right tab, <ss>View Data</ss>.</p>
<ul>
<li>Double click on the job name to open its reports.</li>
<li>Click on different <ss>Reports</ss> in the left hand panel.</li>
</ul>
<p>Things to look at:</p>
<p><strong>General: Filtering (polymerase reads)</strong></p>
<ul>
<li>number of reads post-filter</li>
<li>read length (=average)</li>
</ul>
<p><img alt="filtering results" src="./images/filtering_results.png" /></p>
<p><strong>General: Subread Filtering (subreads)</strong></p>
<ul>
<li>number of reads post-filter</li>
<li>read length (average)</li>
</ul>
<p><img alt="subreads results" src="./images/subreads_results.png" /></p>
<p><strong>Assembly: Pre-Assembly (pre-assembled reads)</strong></p>
<ul>
<li>length cutoff (the computed minimum seed read length)</li>
<li>read length (average)</li>
</ul>
<p><img alt="preassembly results" src="./images/preassembly_results.png" /></p>
<p><strong>Assembly: Corrections</strong></p>
<p>Consensus calling results:</p>
<ul>
<li>Consensus concordance should be &gt; 99%.</li>
</ul>
<p>Graph: corrections across reference:</p>
<ul>
<li>With the first run of polishing, we expect a lot of corrections but they should be randomly distributed.</li>
</ul>
<p><img alt="corrections results" src="./images/corrections_results.png" /></p>
<p><em>Note: only unitigs 0 and 1 shown.</em></p>
<p><strong>Assembly: Top Corrections</strong></p>
<p>This is a list of all the corrections made.</p>
<p><img alt="top corrections results" src="./images/topcorrections_results.png" /></p>
<p><em>Note: only first 15 shown.</em></p>
<!-- - If more than two corrections (with confidence > 50), repeat polishing (see next section "Further polishing"). -->

<p><strong>Resequencing: Coverage</strong></p>
<p>Coverage across reference:</p>
<ul>
<li>discard contigs &lt;20X coverage</li>
<li>others should have fairly consistent coverage.</li>
<li>spikes could be collapsed repeats.</li>
<li>valleys could be mis-assembly - e.g. draft assembly was incorrect and so remapped reads didn&rsquo;t support this part of the assembly.</li>
</ul>
<p><img alt="resequencing coverage results" src="./images/reseq_coverage_results.png" /></p>
<p><em>Note: only unitigs 0 and 1 shown.</em></p>
<p>Graph: Depth of Coverage:</p>
<p>The is the <em>number</em> of reference regions vs. coverage. <!-- Should be roughly normal? --></p>
<p><img alt="depth coverage results" src="./images/depth_coverage_results.png" /></p>
<p><strong>Assembly: Polished Assembly</strong></p>
<ul>
<li>Number of contigs</li>
<li>Max contig length</li>
<li>Graph: confidence vs depth. Multi-copy plasmids may have higher coverage.</li>
</ul>
<p><img alt="polished assembly results" src="./images/polished_assembly_results.png" /></p>
<!--
Sarah Baines: I have found with our data that there is no point in repeating quiver more than once after HGAP, and especially if we are going to polish the assembly with short read data later. A third run has never made a significant improvement to any of the genomes I have worked with, and after a few runs quiver starts to cycle - where is makes a correction then corrects it back to the original call in the subsequent run and so on.

- therefore I have hidden this section for now - AS.

## Further Polishing

During polishing, raw reads are used to correct the assembly.
During HGAP, the assembly was polished once but may need further corrections.

- From the previous step, Go to <ss>Data &rarr; Assembly &rarr; Polished Assembly</ss> and download the FASTA file by clicking on it.
    - Unzip the .gz file
- Go to <ss>Design Job &rarr; Import and Manage</ss> and click <ss>New</ss> on the bottom right hand side. Then, select that FASTA assembly file to upload.
    - creates a new reference.
- Go to <ss>Design Job &rarr; Create New</ss>
    - choose reference-based
    - Select protocol: RS_Resequencing.1
    - Leave all settings.
    - Select your reference from the drop down menu.
    - Click <ss>Save</ss> and <ss>Start</ss>.
- Examine the output assembly and repeat if necessary (e.g. if > 2 corrections with >50 confidence).
-->

<h2 id="output">Output</h2>
<p>The polished assembly as a FASTA file.</p>
<p><img alt="output fasta file" src="./images/output_fasta.png" /></p>
<ul>
<li>download to local computer; or</li>
<li>open file in (GVL) Galaxy; or</li>
<li>open file in GVL command line and perform further analysis.</li>
</ul>
<!-- ## Correct with short reads

If you have Illumina reads for the same sample, the assembly can be further polished.

- Download the FASTA assembly to your computer.
- Open Galaxy
- Upload the FASTA assembly
- Upload the Illumina reads to galaxy
- change datatype to fastqsanger
ngs mapping - map with bwa mem
-  use genome from history - the smrt portal fasta file
R1 and R2 files
execute
refresh

- Polish with Pilon.

pilon
use a genome from history : the smrt portal fasta file
input bam file - from step above
variant calling mode - no
create changes file - yes
advanced options:
paired end : yes
fix - select all
mindepth 0.5



output = polished assembly
-->

<h2 id="next">Next</h2>
<p>Further options:</p>
<ul>
<li>
<p>Circularise the contigs; e.g. with Circlator.</p>
</li>
<li>
<p>Polish the assembly using Illumina data; e.g. with Pilon.</p>
</li>
<li>
<p>Annotate the polished assembly; e.g. with Prokka.</p>
</li>
</ul>
<h2 id="links-to-more-information">Links to more information</h2>
<p><a href="https://github.com/PacificBiosciences/DevNet/wiki/E.-coli-Bacterial-Assembly">PacBio <em>E. coli</em> data set</a></p>
<p><a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/HGAP">HGAP overview</a></p>
<p><a href="http://files.pacb.com/software/smrtanalysis/2.3.0/doc/smrtportal/help/Webhelp/SMRT_Portal.htm">A full ist of reports and terminology</a></p>
<p><a href="http://www.pacb.com/training/BacterialAssemblyandEpigeneticAnalysis/story.html">Video overview of HGAP on SMRT portal</a></p>
<p><a href="http://files.pacb.com/Training/IntroductiontoSMRTbellTemplatePreparation/story_content/external_files/Introduction%20to%20SMRTbell%E2%84%A2%20Template%20Preparation.pdf">More about the SMRT bell template</a></p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../kraken/" title="Sample identification with Kraken" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Sample identification with Kraken
              </span>
            </div>
          </a>
        
        
          <a href="../cmdline_assembly/" title="PacBio assembly with command line tools" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                PacBio assembly with command line tools
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2016-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://omics.data.edu.au/" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application-6b599127bc.js"></script>
      <script>app.initialize({url:{base:"../.."}})</script>
      
        <script src="../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>