<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="../../favicon.ico">
        

	<title>Assembly and finishing with command line tools - ABRPI Training 2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Microbial Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Advanced Microbial Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../pacbio/">Assembly with SMRT portal</a>
</li>

                    
                        
<li class="active">
    <a href="./">Assembly and finishing with command line tools</a>
</li>

                    
                        
<li >
    <a href="../cmdline_prokka/">Annotation using command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Microbial Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../pacbio/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../cmdline_prokka/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#pacbio-reads-assembly-with-command-line-tools">PacBio reads: assembly with command line tools</a></li>
        
            <li><a href="#learning-objectives">Learning objectives</a></li>
        
            <li><a href="#quickstart-overview">Quickstart overview</a></li>
        
            <li><a href="#get-data">Get data</a></li>
        
            <li><a href="#assemble">Assemble</a></li>
        
            <li><a href="#trim-and-circularise">Trim and circularise</a></li>
        
            <li><a href="#find-smaller-plasmids">Find smaller plasmids</a></li>
        
            <li><a href="#correct">Correct</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="pacbio-reads-assembly-with-command-line-tools">PacBio reads: assembly with command line tools</h1>
<p>This tutorial demonstrates how to use long PacBio sequence reads to assemble a bacterial genome and plasmids, including correcting the assembly with short Illumina reads.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial, be able to use command line tools to produce a bacterial genome assembly using the following workflow:</p>
<ol>
<li>get data</li>
<li>assemble long (PacBio) reads</li>
<li>trim overhangs</li>
<li>circularise</li>
<li>search for smaller plasmids</li>
<li>correct with short (Illumina) reads</li>
</ol>
<h2 id="quickstart-overview">Quickstart overview</h2>
<p>A summary of the commands used in this example (not all directory changes and minor steps listed):</p>
<p>(Note, filenames in this section and the next section are not yet finalised).</p>
<p>example with Staph aureus. Sample 25745</p>
<pre><code class="bash"># join files and analyse with canu
cat pacbio* &gt; subreads.fastq.gz
# gunzip
# find info about read lengths
fq subreads.fastq
# average length is 10k; mode is 20k
mkdir 01.canu
cd 01.canu
canu -p canu -d canu.default genomeSize=2.8m -pacbio-raw ../subreads.fastq.gz
# examine how many contigs found and their lengths
cd canu.default/
fa -f canu.contigs.fasta
# 1 chrm; 1 plasmid
# if too many contigs found, repeat canu with sensitive options
# use circlator to trim overhangs and orient start of assemblies
mkdir 02.circlator
cd circlator
# using a length cutoff of approx 2X mode read length
circlator all --threads 72 --verbose --b2r_length_cutoff 40000 ../01.canu/canu.default/canu.contigs.fasta ../01.canu/canu.default/canu.correctedReads.fasta.gz circlator_all_output
# check the output to see if contigs were circularised
cd circlator_all_output
less 04.merge.circularise.log
# check where the contigs were oriented
less 06.fixstart.log
# look at new contig trimmed lengths
fa -f 06.fixstart.fasta
# rename file and copy up
mv 06.fixstart.fasta canu.circlator.contigs.fa
# use illumina reads to find smaller plasmids
mkdir 03.find_smaller_plasmids
bwa index ../canu.circlator.contig.fa
bwa mem -t 32 ../canu.circlator.contig.fa ../R1 ../R2 | samtools sort &gt; aln.bam
samtools index aln.bam
samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq -o spades_assembly
# use sizeseq to find longest contig
# blast start agains all; if overhang found; trim

    # if no overhang found: ncbi blastx hits a rep protein suggesting this is a small plasmid
    # assemble all illumina reads to find flanking regions
    # mkdir 04.all_illumina_spades
    spades-fast --R1 ../R1 --R2 ../R2 --gsize 2.8M --outdir spades_fast --cpus 32
    # view assembly_graph.fastg in bandagae and blast using the contig
    # it matches a node of length 2373: extract this node from the all-illumina contigs
    samtools faidx contigs.fasta
    samtools faidx contigs.fasta NODE_11_length_2373_cov_417.492 &gt; contig3b.fa
    # blast start against all to find overhang
    head -n 10 contig3b.fa &gt; contig3b.fa.head
    makeblastdb -in contig3b.fa -dbtype nucl
    blastn -query contig3b.fa.head -db contig3b.fa -evalue 1e-3 -dust no -out contig3b.bls
    samtools faidx contig3b.fa
    samtools faidx contig3b.fa contig3b:1-2252 &gt; contig3b_trimmed.fa
    # ncbi blast x: find this matches rep protein; save as nucleotides as rep_protein.fa

# fixstart
circlator fixstart --genes_fa rep_protein.fa contig3b_trimmed.fa fixstart
# join all contigs and plasmids
cat canu.circlator.contigs.fa small_plasmid.fa &gt; pre_pilon_staph.fa
#run pilon to correct using illumina reads
mkdir 05.pilon/
cd 05.pilon/
bwa index ../pre_pilon_staph.fa
bwa mem -t 32 ../pre_pilon_staph.fa ../R1 ../R2 | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx ../pre_pilon_staph.fa
pilon --genome ../pre_pilon_staph.fa --frags aln.bam --output pilon --fix all --mindepth 0.5 --changes --threads 32 --verbose
#check the output: one large deletion - this is a tandem repeat in the pacbio assembly; unsupported by the illumina reads.
# final assembly pilon.fasta changed name to staph_25745.fasta

options:

#re-orient larger plasmid, rather that use circlator's choice
samtools faidx staph_25745.fasta
samtools faidx staph_25745.fasta tig00000001_pilon &gt; large.plasmid.fa
samtools faidx staph_25745.fasta tig00000000_pilon &gt; chrm.fa
samtools faidx staph_25745.fasta contig3b:1-2252_pilon &gt; small.plasmid.fa
# blastn on ncbi : matches staph aureus plasmid CP002115.1 by Tim Stinear's group
# this is oriented at start of repA; saved this as a fasta file
circlator fixstart --genes_fa repA_CP002115.fa ../large.plasmid.fa fixstart
# rejoin all three contigs into a multifasta file
</code></pre>

<h2 id="get-data">Get data</h2>
<ul>
<li>Open the mGVL command line</li>
<li>Navigate to or create the directory in which you want to work.
e.g.</li>
</ul>
<pre><code>mkdir staph
cd staph
</code></pre>

<h3 id="find-the-pacbio-files-for-this-sample">Find the PacBio files for this sample</h3>
<ul>
<li>
<p>Obtain the input files. e.g. from the BPA portal.</p>
</li>
<li>
<p>PacBio files are often stored in the format: Sample_name/Cell_name/Analysis_Results/long_file_name_1.fastq.gz</p>
</li>
<li>We will use the <fn>longfilename.subreads.fastq.gz</fn> files.</li>
<li>The reads are usually split into three separate files because they are so large.</li>
<li>Right click on the first <fn>subreads.fastq.gz</fn> file and &ldquo;copy link address&rdquo;.</li>
<li>In the command line, type:</li>
</ul>
<pre><code class="bash">wget --user username --password password [paste link URL for file]
</code></pre>

<ul>
<li>repeat for the other two <fn>subreads.fastq.gz</fn> files.</li>
<li>Alternatively, if the files are already on your server, you can symlink by using</li>
</ul>
<pre><code>ln -s real_file_path chosen_file_path
</code></pre>

<h3 id="join-pacbio-fastq-files">Join PacBio fastq files</h3>
<ul>
<li>If the files are gzipped, type:</li>
</ul>
<pre><code>cat filepath/filep0.*.subreads.fastq.gz &gt; subreads.fastq.gz
</code></pre>

<ul>
<li>If the files are not gzipped, type:</li>
</ul>
<pre><code>cat filepath/filep0.*.subreads.fastq | gzip &gt; subreads.fastq.gz
</code></pre>

<ul>
<li>We now have a file called <fn>subreads.fastq.gz</fn>.</li>
</ul>
<h3 id="find-the-illumina-files-for-this-sample">Find the Illumina files for this sample</h3>
<ul>
<li>We will also use 2 x Illumina (Miseq) fastq.gz files.</li>
<li>These are the <fn>R1.fastq.gz</fn> and <fn>R2.fastq.gz</fn> files.</li>
<li>Right click on the file name and &ldquo;copy link address&rdquo;.</li>
<li>In the command line, type:</li>
</ul>
<pre><code>wget --user username --password password [paste link URL for file]
</code></pre>

<ul>
<li>Repeat for the other read.fastq.gz file.</li>
<li>Shorten the name of each of these files:</li>
</ul>
<pre><code>mv longfilename_R1.fastq.gz R1.fastq.gz
mv longfilename_R2.fastq.gz R2.fastq.gz
</code></pre>

<h3 id="view-files">View files</h3>
<ul>
<li>Type &ldquo;ls&rdquo; to display the folder contents.</li>
</ul>
<pre><code>ls
</code></pre>

<ul>
<li>The 3 files we will use in this analysis are:<ul>
<li><fn>subreads.fastq.gz</fn> (the PacBio reads)</li>
<li><fn>R1.fastq.gz</fn> and <fn>R2.fastq.gz</fn> (the Illumina reads)</li>
</ul>
</li>
</ul>
<h2 id="assemble">Assemble</h2>
<ul>
<li>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>.</li>
<li>Run canu with these commands:</li>
</ul>
<pre><code>canu -p staph -d output genomeSize=2.8m -pacbio-raw subreads.fastq
</code></pre>

<ul>
<li><strong>staph</strong> is the prefix given to output files</li>
<li><strong>output</strong> is the output directory</li>
<li><strong>genomeSize</strong> only has to be approximate.<ul>
<li>e.g. <em>Staphylococcus aureus</em>, 2.8m</li>
<li>e.g. <em>Streptococcus pyogenes</em>, 1.8m</li>
</ul>
</li>
<li>the <strong>reads</strong> can be unzipped or .gz</li>
</ul>
<!--
corMhapSensitivity=high corMinCoverage=0
-->

<!--
- Note: it may say "Finished..."  but it is probably still running. Type:

wzxhzdk:10

- This will show you what is running.
-->

<ul>
<li>Canu will correct, trim and assemble the reads.</li>
<li>This will take ~ 30 minutes.</li>
</ul>
<h3 id="check-the-output">Check the output</h3>
<pre><code>cd output
</code></pre>

<ul>
<li>The <fn>staph.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>staph.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>staph.correctedReads.fasta.gz</fn> are the corrected pacbio reads that were used in the assembly.</li>
<li>The <fn>staph.file.gfa</fn> is the graph of the assembly.</li>
<li>Display summary information about the contigs:</li>
</ul>
<pre><code>fa -f staph.contigs.fasta
</code></pre>

<ul>
<li>This will show the number of contigs.</li>
</ul>
<p>[add screenshot]</p>
<p>e.g.</p>
<p>tig00000000 dna 2746242</p>
<p>tig00000001 dna 48500</p>
<p>(stdin) no=2 bp=2794742 ok=2794742 Ns=0 gaps=0 min=48500 avg=1397371 max=2746242 N50=2746242</p>
<h3 id="change-canu-parameters-if-required">Change canu parameters if required</h3>
<ul>
<li>If the assembly is poor with many contigs, re-run canu with extra sensitivity parameters; e.g.</li>
</ul>
<pre><code>canu -p prefix -d outdir corMhapSensitivity=high corMinCoverage=0 genomeSize=2.8m -pacbio-raw subreads.fastq
</code></pre>

<h2 id="trim-and-circularise">Trim and circularise</h2>
<h3 id="run-circlator">Run circlator</h3>
<pre><code>circlator all --threads 72 --verbose ../staph.contigs.fasta ../staph.corrected.reads.fastq.gz circlator_all_output
</code></pre>

<p>Check the output: contig sizes, whether they were circularised, trimmed and the start position chosen.</p>
<!--
###Separate the contigs into single files
- Index the contigs file:

wzxhzdk:15

- this makes an indexed file with the suffix -fai
- send each contig to a new file:

wzxhzdk:16

- change contig names:

wzxhzdk:17

- change header to >contig1 or >contig2
- save
- We now have two files:
    - <fn>contig1.fa</fn> : the chromosome
    - <fn>contig2.fa</fn> : a plasmid

-->

<!-- ###What is in the unassembled reads?

- Are they contaminants?
- Blast against NCBI
    - Use Cyberduck to transfer the file to your local computer
    - Go to NCBI and upload the file
    - (doesn't work with this file - too big)
    - (or: explain how to do so on command line?)
-->

<!--## Alternatives for assembly
- canu without sensitivity settings
- HGAP2

##Use Quiver/Arrow here?

to correct assembly with the raw reads.
-->

<!---
## Trim - manuallly

The bacterial chromosome and plasmids are circular.

The canu assembly graph will be split at an ambiguous/unstable node. However, this area of the graph likely overlaps in the bacterial chromosome, but has not aligned with itself completely. This is called an overhang. We need to identify these overhangs and trim them, for the chromosome and any plamsids.
### Chromosome: identify overhang
- Take the first 30,000 bases of contig1.

wzxhzdk:18

- this is the start of the assembly
- we want to see if it matches the end (overhang)
- format the assembly file for blast:

wzxhzdk:19

- blast the start of the assembly (.head file) against all of the assembly:

wzxhzdk:20

- look at <fn>contig1.bls</fn> to see hits:

wzxhzdk:21

- The first hit is against the start of the chromosome, as expected.
- To find the next hit, type:

wzxhzdk:22

then

wzxhzdk:23

to scroll through the remaining hits.

- The next hit is from position 2,725,223 until the end of the chromosome (position 2,746,242).

![screeshot of blast](images/blast_chrm.png)

- Therefore, the overhang starts at position 2,725,233 and we can trim the contig to this position minus 1.
### Chromosome: trim
- First, index the <fn>contig1.fa</fn> file

wzxhzdk:24

- this makes an index file called <fn>contig1.fa.fai</fn>
- next, extract all the sequence except for the overhang. (We don't have to specify the name of the *index* file; it will be found automatically):

wzxhzdk:25

- open the <fn>contig1.fa.trimmed</fn> file

wzxhzdk:26

- delete the header info except contig name (e.g. contig1)
- save, exit.
- we now have a trimmed contig1.
### Plasmid: examine
- Contig 2 is 48,500 bases.
- This seems long for a plasmid in this species.
- Do a dot plot to examine the sequence.
###Transfer file to local computer
- Use Cyberduck to copy <fn>contig2.fa</fn> to your local computer
    - Install Cyberduck
    - Open Cyberduck
    - click on "open connection"
    - choose SFTP from drop down menu
    - server = your virtual machine IP address (e.g. abrpi.genome.edu.au)
    - username = your username
    - password = your password
    - You should now see a window showing the folders and files on your virtual machine.
    - You can drag and drop files into the preferred folder.
###View dot plot
- Install Gepard
- Open Gepard
- Sequences - sequence 1 - select file - <fn>contig2.fa</fn>
- Sequences - sequence 2 - select file - <fn>contig2.fa</fn>
- Create dotplot
![Gepard dotplot screenshot](images/gepard_contig2.png)
- the sequence starts to repeat at position 24880
- there are probably two plasmids combined into one contig in tandem.
- the whole length is 48500
- we will cut 10,000 from each end

###Trim plasmid
- use samtools to extract the region of the contig except for 10k on each end

wzxhzdk:27

- we now have a ~ 28500 bases plasmid

wzxhzdk:28

- delete the rest of the name after >contig2
###Trim overhang in the plasmid

- Take the first x bases:

wzxhzdk:29

- this is the start of the assembly
- we want to see if it matches the end (overhang)
- format the assembly file for blast:

wzxhzdk:30

- blast the start of the assembly against all of the assembly:

wzxhzdk:31

- look at contig2.bls to see hits:

wzxhzdk:32

- The first hit is against the start of the chromosome, as expected.
- The last hit starts at position 24886.
- We will trim the plasmid to position 24885
- first, index the contig2.fa.half file

wzxhzdk:33

- trim

wzxhzdk:34

- open the contig2.fa.half.trimmed file

wzxhzdk:35

- delete the header info except contig name (e.g. contig2)
exit.
- we now have a trimmed contig2.
--->

<h2 id="find-smaller-plasmids">Find smaller plasmids</h2>
<p><strong>Optional advanced section</strong></p>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the multifasta canu-circlator output of trimmed assembly contigs.</li>
<li>Map all the Illumina reads against these pacbio assembled contigs.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: none found.</li>
<li>Search Genbank for any matching proteins: a replication protein found.  </li>
<li>Assemble all the Illumina reads and produce an assembly graph.</li>
<li>Search the graph for a match to the replication protein and its adjoining regions.</li>
<li>Extract this longer sequence from the Illumina assembly: this is the small plasmid.</li>
<li>Check for overhang in this plasmid and trim.</li>
</ol>
<!-- combine contigs 1 and 2
take chr and plasmid => one fasta file

wzxhzdk:36


wzxhzdk:37

- Make directory and move in combined contigs file and the illumina reads

wzxhzdk:38

-->

<h3 id="align-illumina-with-bwa">Align Illumina with BWA</h3>
<ul>
<li>Align illumina reads to these contigs</li>
<li>First, index the contigs file</li>
</ul>
<pre><code>bwa index contig_1_2.fa
</code></pre>

<ul>
<li>then, align using bwa mem</li>
</ul>
<pre><code>bwa mem -t 8 contig_1_2.fa R1.fastq.gz R2.fastq.gz | samtools sort &gt; aln.bam
</code></pre>

<ul>
<li>the output alignment is <fn>aln.bam</fn></li>
</ul>
<!--
###Or align with bowtie2
-->

<h3 id="extract-unmapped-illlumina-reads">Extract unmapped illlumina reads</h3>
<ul>
<li>Index the alignment file</li>
</ul>
<pre><code>samtools index aln.bam
</code></pre>

<ul>
<li>extract the fastq files from the bam alignment - those reads that were unmapped to the pacbio alignment - and save them in various &ldquo;unampped&rdquo; files:</li>
</ul>
<pre><code>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</code></pre>

<!--
- check R1 and R2 are approx same size:

wzxhzdk:43

- check first reads have same name in R1 and R2:

wzxhzdk:44

-->

<ul>
<li>we now have three files of the unampped reads:</li>
<li><fn> unmapped.R1.fastq</fn></li>
<li><fn> unmapped.R2.fastq</fn></li>
<li><fn> unmapped.RS.fastq</fn></li>
</ul>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads</h3>
<ul>
<li>assemble with spades</li>
</ul>
<pre><code>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq -o spades_assembly
</code></pre>

<p>-1 is input file forward
-2 is input file reverse
-s is unpaired
-o is output_dir</p>
<pre><code>cd spades_assembly
fa contigs.fasta
</code></pre>

<ul>
<li>shows how many assembled:<ul>
<li>e.g. no=135</li>
<li>max = 2229</li>
</ul>
</li>
<li>sort fasta by size of seqs:</li>
</ul>
<pre><code>sizeseq
</code></pre>

<p>Input sequence set: contigs.fasta
Return longest sequence first [N]: Y
output sequence(s) [contigs.fasta]: sorted_contigs.fasta
just print the first row of each seq to see coverage:</p>
<pre><code>grep cov sorted_contigs.fasta  
</code></pre>

<ul>
<li>result: NODE_1_length_2229_cov_610.583<ul>
<li>longest contig is 2229 and high coverage</li>
</ul>
</li>
<li>all the nodes are listed</li>
<li>see if any other ones have high coverage<ul>
<li>e.g. &gt;NODE_135_length_78_cov_579</li>
</ul>
</li>
<li>look at the sequence of this contig:</li>
</ul>
<pre><code>tail sorted_contigs.fasta
</code></pre>

<ul>
<li>seq looks bad for node135 so disregard (seq is only Cs)</li>
<li>we will extract the first seq</li>
</ul>
<pre><code>samtools faidx sorted_contigs.fasta
samtools faidx sorted_contigs.fasta NODE_1_length_2229_cov_610.583 &gt; contig3.fa
</code></pre>

<ul>
<li>this is now saved as <fn>contig3.fa</fn></li>
<li>nano, make header &gt;contig3, save.</li>
</ul>
<h3 id="investigate-the-small-plasmid-contig3">Investigate the small plasmid (contig3)</h3>
<ul>
<li>blast start of contig3 against itself</li>
<li>Take the first x bases:</li>
</ul>
<pre><code>head -n 10 contig3.fa &gt; contig3.fa.head
</code></pre>

<ul>
<li>this is the start of the assembly</li>
<li>we want to see if it matches the end (overhang)</li>
<li>format the assembly file for blast:</li>
</ul>
<pre><code>makeblastdb -in contig3.fa -dbtype nucl
</code></pre>

<ul>
<li>blast the start of the assembly (.head file) against all of the assembly:</li>
</ul>
<pre><code>blastn -query contig3.fa.head -db contig3.fa -evalue 1e-3 -dust no -out contig3.bls
</code></pre>

<ul>
<li>look at <fn>contig3.bls</fn> to see hits:</li>
</ul>
<pre><code>less contig3.bls
</code></pre>

<ul>
<li>the first hit is against itself, as expected</li>
<li>there are no few further hits, so we assume there is no overhang that needs trimming.</li>
<li>however, the sequence is likely then to be longer than this.</li>
<li>copy the sequence:</li>
</ul>
<pre><code>less contig3.fa
</code></pre>

<ul>
<li>copy the sequence</li>
<li>go to ncbi blast: blastx: nuc to <em>protein</em></li>
<li>paste seq</li>
<li>choose genetic code = 11</li>
<li>blast
<img alt="ncbi blast" src="./images/ncbi_blast_contig3.png" /></li>
<li>hits: replication protein</li>
<li>so this is a protein that has not been found in the pacbio assembly.</li>
<li>hypothesise that  this might be a true small plasmid but the rest of its seq is in common with other parts of the staph genome, so they haven&rsquo;t been assembled with the rep protein</li>
</ul>
<h3 id="assemble-all-the-illumina-reads">Assemble <em>all</em> the illumina reads</h3>
<ul>
<li>assemble all the illumina reads with spades (not just those reads unmapped to pacbio assembly)</li>
<li>resulting assembly: search for small plasmid sequence</li>
<li>use spades-fast on MDU server</li>
<li>cd to main analysis folder</li>
<li>mkdir all_illumina_assembly</li>
<li>copy in R1 and R2 and cd in</li>
<li>run spades-fast:</li>
</ul>
<pre><code>spades-fast --R1 R1.fastq.gz --R2 R2.fastq.gz --gsize 2.8M --outdir spades_fast --cpus 32
</code></pre>

<pre><code>cd spades_fast
</code></pre>

<ul>
<li>in here is the <fn>assembly_graph.fastg</fn></li>
<li>use cyberduck or another file transer program to transfer this file to local computer</li>
<li>Examine the assembly in the program Bandage.<ul>
<li>Install Bandage.</li>
<li>File: Load graph: <fn>file.gfa</fn></li>
<li>In the left hand panel, click &ldquo;Draw graph&rdquo;
<img alt="bandage pic" src="./images/illumina_assembly_bandage.png" /></li>
</ul>
</li>
<li>see: main chromosome, plasmid, small plasmid (annotate pic)</li>
<li>blast the small plasmid sequence in this assembly</li>
<li>left hand panel: Blast: create/view BLAST search</li>
<li>build blast database</li>
<li>paste in nuc seq of contig3</li>
<li>blast</li>
<li>the main hit is around node 10</li>
</ul>
<!--- - there are two hits
this is probably due to overhang? eg this is where the plasmid overhang occurs and this is where it would later be trimmed?
--->

<ul>
<li>go to bandage window</li>
<li>&ldquo;find nodes&rdquo; in right hand panel - 10</li>
<li>this node is slightly longer: 2373</li>
<li>this could be the plasmid</li>
<li>go to the folder for spades_fast</li>
</ul>
<pre><code>less contigs.fasta
/2373
</code></pre>

<ul>
<li>search for length of this node</li>
<li>see that this is actually called NODE_11_length_2373_cov_417.492</li>
<li>extract out node 11</li>
</ul>
<pre><code>samtools faidx contigs.fasta
samtools faidx contigs.fasta NODE_11_length_2373_cov_417.492 &gt; contig3b.fa
</code></pre>

<ul>
<li>we will call it contig 3 &ldquo;b&rdquo; because it is larger than our original contig3.</li>
<li>nano, open, change name to contig3b, save</li>
</ul>
<h3 id="trim-small-plasmid">Trim small plasmid</h3>
<ul>
<li>next: check for overhang</li>
<li>Take the first x bases:</li>
</ul>
<pre><code>head -n 10 contig3b.fa &gt; contig3b.fa.head
</code></pre>

<ul>
<li>this is the start of the assembly</li>
<li>we want to see if it matches the end (overhang)</li>
</ul>
<pre><code>makeblastdb -in contig3b.fa -dbtype nucl
blastn -query contig3b.fa.head -db contig3b.fa -evalue 1e-3 -dust no -out contig3b.bls
less contig3b.bls
</code></pre>

<ul>
<li>The first hit is against the start of the chromosome, as expected.</li>
<li>The last hit starts at position 2253</li>
<li>We will trim the plasmid to position 2252</li>
<li>first, index the contig3b.fa file</li>
</ul>
<pre><code>samtools faidx contig3b.fa
</code></pre>

<ul>
<li>trim</li>
</ul>
<pre><code>samtools faidx contig3b.fa contig3b:1-2252 &gt; contig3b.fa.trimmed
</code></pre>

<ul>
<li>open the contig3b.fa.trimmed</li>
</ul>
<pre><code>nano contig3b.fa.trimmed
</code></pre>

<ul>
<li>delete the header info except contig name (e.g. contig3b)
exit.</li>
<li>we now have a trimmed contig3b.</li>
</ul>
<h3 id="collect-all-contigs-in-one-file">Collect all contigs in one file</h3>
<ul>
<li>move up to main analysis folder</li>
<li>mkdir pilon</li>
<li>copy the trimmed contigs 1, 2, 3b into this folder</li>
<li>cat them</li>
</ul>
<pre><code>cat contig_1_2.fa contig3b.fa.trimmed &gt; all_contigs.fa
fa -f all_contigs.fa
</code></pre>

<ul>
<li>see the three contigs and sizes</li>
<li>rename all the contigs</li>
<li>contig1</li>
<li>contig2</li>
<li>contig3b</li>
</ul>
<h2 id="correct">Correct</h2>
<p><!-- check: Canu doesn't do any polishing, so there will be thousands of errors, almost all insertions. -->
We will use the illumina reads to correct the pacbio assembly.</p>
<ul>
<li>inputs:<ul>
<li>draft pacbio assembly (overhang trimmed from each of the three replicons)</li>
<li>illumina reads (aligned to pacbio assembly: in bam format)</li>
</ul>
</li>
<li>output: corrected assembly</li>
</ul>
<h3 id="correct-with-pacbio-corrected-reads">Correct with pacbio corrected reads</h3>
<ul>
<li>to add here.</li>
<li>use the corrected assembly for the next step.</li>
</ul>
<h3 id="align-illumina-reads-bam">Align Illumina reads =&gt; bam</h3>
<p>move the illumina reads into the pilon folder</p>
<pre><code>bwa index all_contigs.fa
bwa mem -t 32 all_contigs.fa R1.fastq.gz R2.fastq.gz | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx all_contigs.fa
</code></pre>

<ul>
<li>-t is the number of cores (e.g. 8)</li>
<li>to find out how many you have, grep -c processor /proc/cpuinfo</li>
<li>now we have an alignment file to use in pilon</li>
<li>look at how the illumina reads are aligned:</li>
</ul>
<pre><code>samtools tview -p contig1 aln.bam all_contigs.fa
</code></pre>

<h3 id="correct_1">Correct</h3>
<!--
contigs & bam => pilon => corrected contigs
don't run on things with lots (10+) of contigs, but 1 or 2 is ok
-->

<ul>
<li>run pilon:</li>
</ul>
<pre><code>pilon --genome all_contigs.fa --frags aln.bam --output corrected --fix bases --mindepth 0.5 --changes --threads 32 --verbose
</code></pre>

<ul>
<li>look at the changes file</li>
</ul>
<pre><code>less corrected.changes
</code></pre>

<ul>
<li>look at the .fasta file.</li>
<li>option: re run pilon to correct again.<ul>
<li>but this time, the reference is the first pilon correction</li>
</ul>
</li>
</ul>
<!--
-but there are some bigger ones.
    - e.g. at position 2,463,699 there is a 23-bp seq that is deleted.

###View the deletion in a pileup

tig00000000:2463600-2463622 tig00000000_pilon:2463699 GTTAAAGGTTATTTGAATGATCA .

- view the illumina reads mapped against the original assembly:

wzxhzdk:70


- -p gives the chromosome position that we want to view
- then input file; then reference file

view:
- reference sequence along the top
- then consensus from the aligned illumina reads
- a dot is a match on F
- a comma is a match on R
- capital letter is a correction on F
- small letter is a correction on R
- asterisk
- underlining

view in IGV:
- transfer aln.bam, aln.bam.bai, contigs.fa, contigs.fa.fai to local computer
- open IGV
- File - genomes - load genome from file: contigs
- File - load from file - aln.bam file
- select the main chromosome eg tig00000000
- go to coordinate 2463600
- there is a dip in coverage here.
- pilon has identified this as a local misassembly and has deleted this section in the corrected assembly

![IGV screenshot](images/IGV_first_pilon_correction.png)

- output:corrected assembly

- re run pilon to correct again.
    - but this time, the reference is the first pilon correction


wzxhzdk:71


- separate into three fasta files

-->

<!--
###or, use Bowtie for alignment

index the fasta file


wzxhzdk:72


input file
output ref name

align:


wzxhzdk:73


- "bowtie" means use the index files that were generated above that we called bowtie
- use 72 threads on mdu but <8 on sepsis
- then index the bam file

wzxhzdk:74


- then view in tview

wzxhzdk:75

<!--
###or use BLASR for alignment: pacbio reads to pacbio assembly

(in MDU marvin)

wzxhzdk:76


blasr subreads.fastq all_contigs.fa -nproc 72 -sam | samtools sort > blasr_aln.bam
samtools index blasr_aln.bam
samtools faidx all_contigs.fa
samtools tview -p tig00000000:2463600 blasr_aln.bam all_contigs.fa

###or align pacbio reads to pacbio assembly with bwa mem

canu: pacbio corrected reads
bwa mem
align to pacbio assembly


bam index all_contigs.fasta

bwa mem -t 72 all_contigs.fa staph.correctedReads.fasta | samtools sort > pacbio_aln.bam


(leave out -x pacbio option as these are corrected reads)
--->

<!--

##Examine 23bp

align corrected pacbio reads to pacbio assembly

make a 23bp seq


wzxhzdk:77


paste in the seq

-->

<!--

### Split into three final contigs

- Examine the all_contigs.fa file

wzxhzdk:78

- contig1   dna 2725158
- contig2   dna 24884
- contig3b  dna 2252
- split these

wzxhzdk:79


##Circularise
corrected contigs => circlator => circular genome (e.g. starting at dnaA)

###Chromosome


wzxhzdk:80


circlator fixstart chromosome.fa chrm



default: orients at DNAa

=> circular, corrected assembly

###Plasmid 1

blast against ncbi
find hit
what have they done
download the start of their assembly (may be a gene or may be tandem repeats etc. )


wzxhzdk:81



 -or, whatever has been done with the plasmid it most closely blast matches to
eg
  https://www.ncbi.nlm.nih.gov/nucleotide/260066114?report=genbank&log$=nuclalign&blast_rank=1&RID=WTK2RJSZ014


need to download

###Plasmid 2

-->

<h3 id="further-analyses">Further analyses</h3>
<p>Annotate with prokka</p>
<h3 id="links">Links</h3>
<p><a href="http://canu.readthedocs.io/en/stable/quick-start.html">Canu manual</a></p>
<p><a href="https://github.com/marbl/canu">Canu code</a></p>
<p><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0112963">Pilon article</a></p>
<p><a href="https://github.com/broadinstitute/pilon/wiki">Pilon on github</a></p>
<p>Circlator
http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0849-0
http://sanger-pathogens.github.io/circlator/</p>
<p>Finishing</p>
<p>https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Finishing-Bacterial-Genomes</p>
<p>https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Evaluating-Assemblies</p>
<p>Files</p>
<p>reference to bas.h5 files (details) https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>