<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="../../favicon.ico">
        

	<title>Assembly and finishing with command line tools - ABRPI Training 2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Microbial Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Advanced Microbial Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../pacbio/">Assembly with SMRT portal</a>
</li>

                    
                        
<li class="active">
    <a href="./">Assembly and finishing with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Microbial Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../pacbio/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../dge/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#pacbio-reads-assembly-with-command-line-tools">PacBio reads: assembly with command line tools</a></li>
        
            <li><a href="#learning-objectives">Learning objectives</a></li>
        
            <li><a href="#get-data">Get data</a></li>
        
            <li><a href="#assemble">Assemble</a></li>
        
            <li><a href="#trim-and-circularise">Trim and circularise</a></li>
        
            <li><a href="#find-smaller-plasmids">Find smaller plasmids</a></li>
        
            <li><a href="#correct">Correct</a></li>
        
            <li><a href="#next">Next</a></li>
        
            <li><a href="#commands-summary">Commands summary </a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="pacbio-reads-assembly-with-command-line-tools">PacBio reads: assembly with command line tools</h1>
<p>This tutorial demonstrates how to use long PacBio sequence reads to assemble a bacterial genome and plasmids, including correcting the assembly with short Illumina reads. The last section contains a <a href="#summary">summary</a> of the main commands used.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial, be able to use command line tools to produce a bacterial genome assembly using the following workflow:</p>
<ol>
<li>get data</li>
<li>assemble long (PacBio) reads</li>
<li>trim overhangs and circularise</li>
<li>search for smaller plasmids</li>
<li>correct with short (Illumina) reads</li>
</ol>
<h2 id="get-data">Get data</h2>
<ul>
<li>Open the mGVL command line</li>
<li>Navigate to or create the directory in which you want to work.
e.g.</li>
</ul>
<pre><code>mkdir staph
cd staph
</code></pre>

<h3 id="find-the-pacbio-files-for-this-sample">Find the PacBio files for this sample</h3>
<ul>
<li>If the files are already on your server, you can symlink by using</li>
</ul>
<pre><code>ln -s real_file_path chosen_symlink_name
</code></pre>

<ul>
<li>Alternatively, obtain the input files from elsewhere, e.g. from the BPA portal. (You will need a password.)</li>
<li>PacBio files are often stored in the format: Sample_name/Cell_name/Analysis_Results/long_file_name_1.fastq.gz</li>
<li>We will use the <fn>longfilename.subreads.fastq.gz</fn> files.</li>
<li>The reads are usually split into three separate files because they are so large.</li>
<li>Right click on the first <fn>subreads.fastq.gz</fn> file and &ldquo;copy link address&rdquo;.</li>
<li>In the command line, type:</li>
</ul>
<pre><code class="bash">wget --user username --password password [paste link URL for file]
</code></pre>

<ul>
<li>Repeat for the other two <fn>subreads.fastq.gz</fn> files.</li>
</ul>
<h3 id="join-pacbio-fastq-files">Join PacBio fastq files</h3>
<ul>
<li>If the files are gzipped, type:</li>
</ul>
<pre><code>cat filepath/filep0.*.subreads.fastq.gz &gt; subreads.fastq.gz
</code></pre>

<ul>
<li>If the files are not gzipped, type:</li>
</ul>
<pre><code>cat filepath/filep0.*.subreads.fastq | gzip &gt; subreads.fastq.gz
</code></pre>

<ul>
<li>We now have a file called <fn>subreads.fastq.gz</fn>.</li>
</ul>
<h3 id="find-the-illumina-files-for-this-sample">Find the Illumina files for this sample</h3>
<ul>
<li>We will also use 2 x Illumina (Miseq) fastq.gz files.</li>
<li>These are the <fn>R1.fastq.gz</fn> and <fn>R2.fastq.gz</fn> files.</li>
<li>Right click on the file name and &ldquo;copy link address&rdquo;.</li>
<li>In the command line, type:</li>
</ul>
<pre><code>wget --user username --password password [paste link URL for file]
</code></pre>

<ul>
<li>Repeat for the other read.fastq.gz file.</li>
<li>Shorten the name of each of these files:</li>
</ul>
<pre><code>mv longfilename_R1.fastq.gz R1.fastq.gz
mv longfilename_R2.fastq.gz R2.fastq.gz
</code></pre>

<h3 id="view-files">View files</h3>
<ul>
<li>Type &ldquo;ls&rdquo; to display the folder contents.</li>
</ul>
<pre><code>ls
</code></pre>

<ul>
<li>
<p>The 3 files we will use in this analysis are:</p>
<ul>
<li><fn>subreads.fastq.gz</fn> (the PacBio reads)</li>
<li><fn>R1.fastq.gz</fn> and <fn>R2.fastq.gz</fn> (the Illumina reads)</li>
</ul>
</li>
<li>
<p>In this tutorial we will use <em>Staphylococcus aureus</em> sample 25745.</p>
</li>
</ul>
<h2 id="assemble">Assemble</h2>
<ul>
<li>We will use the assembly software called <a href="http://canu.readthedocs.io/en/stable/">Canu</a>.</li>
<li>Run Canu with these commands:</li>
</ul>
<pre><code>canu -p staph -d output genomeSize=2.8m -pacbio-raw subreads.fastq
</code></pre>

<ul>
<li><strong>staph</strong> is the prefix given to output files</li>
<li><strong>output</strong> is the name of the output directory</li>
<li><strong>genomeSize</strong> only has to be approximate.<ul>
<li>e.g. <em>Staphylococcus aureus</em>, 2.8m</li>
<li>e.g. <em>Streptococcus pyogenes</em>, 1.8m</li>
</ul>
</li>
<li>
<p>the <strong>reads</strong> can be unzipped or .gz</p>
</li>
<li>
<p>Canu will correct, trim and assemble the reads.</p>
</li>
<li>This will take ~ 30 minutes.</li>
<li>Type</li>
</ul>
<pre><code>squeue
</code></pre>

<p>to check that the job is running.</p>
<h3 id="check-the-output">Check the output</h3>
<pre><code>cd output
</code></pre>

<ul>
<li>The <fn>staph.contigs.fasta</fn> are the assembled sequences.</li>
<li>The <fn>staph.unassembled.fasta</fn> are the reads that could not be assembled.</li>
<li>The <fn>staph.correctedReads.fasta.gz</fn> are the corrected pacbio reads that were used in the assembly.</li>
<li>The <fn>staph.file.gfa</fn> is the graph of the assembly.</li>
<li>Display summary information about the contigs:</li>
</ul>
<pre><code>fa -f staph.contigs.fasta
</code></pre>

<ul>
<li>This will show the number of contigs, e.g.<ul>
<li>tig00000000   dna 2746242</li>
<li>tig00000001   dna 48500</li>
</ul>
</li>
</ul>
<h3 id="change-canu-parameters-if-required">Change Canu parameters if required</h3>
<ul>
<li>If the assembly is poor with many contigs, re-run Canu with extra sensitivity parameters; e.g.</li>
</ul>
<pre><code>canu -p prefix -d outdir corMhapSensitivity=high corMinCoverage=0 genomeSize=2.8m -pacbio-raw subreads.fastq
</code></pre>

<h2 id="trim-and-circularise">Trim and circularise</h2>
<h3 id="run-circlator">Run Circlator</h3>
<p>Circlator identifies and trims overhangs (on chromosomes and plasmids) and orients the start position at an appropriate gene (e.g. dnaA). It takes in the assembled contigs from Canu, as well as the corrected reads prepared by Canu.</p>
<p>To run:</p>
<pre><code>circlator all --verbose ../staph.contigs.fasta ../staph.corrected.reads.fastq.gz circlator_all_output
</code></pre>

<ul>
<li>
<p><strong>- -verbose</strong> prints progress information to the screen</p>
</li>
<li>
<p><strong>../staph.contigs.fasta</strong> is the file path to the input multi-fasta assembly</p>
</li>
<li>
<p><strong>../staph.corrected.reads.fastq.gz</strong> is the file path to the corrected pacbio reads</p>
</li>
<li>
<p><strong>circlator_all_output</strong> is the name of the output directory.</p>
</li>
</ul>
<p>Check the output: contig sizes, whether they were circularised, trimmed and the start position chosen.</p>
<pre><code>less 04.merge.circularise.log
less 06.fixstart.log
</code></pre>

<p>The trimmed contigs are in the file called <fn>06.fixstart.fasta</fn>. Re-name it <fn>contig_1_2.fa</fn>:</p>
<pre><code>mv 06.fixstart.fasta contig_1_2.fa
</code></pre>

<h2 id="find-smaller-plasmids">Find smaller plasmids</h2>
<p>Pacbio reads are long, and may have been longer than small plasmids. We will look for any small plasmids using the Illumina reads.</p>
<p>This section involves several steps:</p>
<ol>
<li>Use the multifasta canu-circlator output of trimmed assembly contigs.</li>
<li>Map all the Illumina reads against these pacbio assembled contigs.</li>
<li>Extract any reads that <em>didn&rsquo;t</em> map and assemble them together: this could be a plasmid, or part of a plasmid.</li>
<li>Look for overhang: if found, trim. If not, continue:</li>
<li>Search Genbank for any matching proteins: a replication protein found.  </li>
<li>Assemble all the Illumina reads and produce an assembly graph.</li>
<li>Search the graph for a match to the replication protein and its adjoining regions.</li>
<li>Extract this longer sequence from the Illumina assembly: this is the small plasmid.</li>
<li>Check for overhang in this plasmid and trim.</li>
</ol>
<h3 id="align-illumina-with-bwa">Align Illumina with BWA</h3>
<ul>
<li>Align illumina reads to these contigs</li>
<li>First, index the contigs file</li>
</ul>
<pre><code>bwa index contig_1_2.fa
</code></pre>

<ul>
<li>then, align using bwa mem</li>
</ul>
<pre><code>bwa mem -t 8 contig_1_2.fa R1.fastq.gz R2.fastq.gz | samtools sort &gt; aln.bam
</code></pre>

<ul>
<li><strong>bwa mem</strong> is the alignment tool</li>
<li><strong>-t 8</strong> is the number of cores</li>
<li><strong>contig_1_3.fa</strong> is the input assembly file</li>
<li><strong>R1.fastq.gz R2.fastq.gz</strong> are the Illumina reads</li>
<li><strong> | samtools sort</strong> pipes the output to samtools to sort</li>
<li><strong>&gt; aln.bam</strong> sends the alignment to the file <fn>aln.bam</fn></li>
</ul>
<h3 id="extract-unmapped-illlumina-reads">Extract unmapped illlumina reads</h3>
<ul>
<li>Index the alignment file</li>
</ul>
<pre><code>samtools index aln.bam
</code></pre>

<ul>
<li>extract the fastq files from the bam alignment - those reads that were unmapped to the pacbio alignment - and save them in various &ldquo;unmapped&rdquo; files:</li>
</ul>
<pre><code>samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam
</code></pre>

<ul>
<li>we now have three files of the unampped reads:</li>
<li><fn> unmapped.R1.fastq</fn></li>
<li><fn> unmapped.R2.fastq</fn></li>
<li><fn> unmapped.RS.fastq</fn></li>
</ul>
<h3 id="assemble-the-unmapped-reads">Assemble the unmapped reads</h3>
<ul>
<li>assemble with spades</li>
</ul>
<pre><code>spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq -o spades_assembly
</code></pre>

<ul>
<li><strong>-1</strong> is input file forward</li>
<li><strong>-2</strong> is input file reverse</li>
<li><strong>-s</strong> is unpaired</li>
<li><strong>-o</strong> is the output directory</li>
</ul>
<pre><code>cd spades_assembly
fa contigs.fasta
</code></pre>

<ul>
<li>shows how many assembled:<ul>
<li>e.g. no=135</li>
<li>max = 2229</li>
</ul>
</li>
<li>sort fasta by size of seqs:</li>
</ul>
<pre><code>sizeseq
</code></pre>

<ul>
<li>Input sequence set: contigs.fasta</li>
<li>Return longest sequence first [N]: Y</li>
<li>output sequence(s) [contigs.fasta]: sorted_contigs.fasta</li>
</ul>
<p>Print the first row of each seq to see coverage:</p>
<pre><code>grep cov sorted_contigs.fasta  
</code></pre>

<ul>
<li>result: NODE_1_length_2229_cov_610.583<ul>
<li>longest contig is 2229 and high coverage</li>
</ul>
</li>
<li>all the nodes are listed</li>
<li>see if any other nodes have high coverage<ul>
<li>e.g. NODE_135_length_78_cov_579</li>
</ul>
</li>
<li>look at the sequence of this contig:</li>
</ul>
<pre><code>tail sorted_contigs.fasta
</code></pre>

<ul>
<li>This is a homopolymer, so disregard.</li>
<li>We will extract the first sequence (NODE_1):</li>
</ul>
<pre><code>samtools faidx sorted_contigs.fasta
samtools faidx sorted_contigs.fasta NODE_1_length_2229_cov_610.583 &gt; contig3.fa
</code></pre>

<ul>
<li>this is now saved as <fn>contig3.fa</fn></li>
<li>open this file in nano, make the header &ldquo;&gt;contig3&rdquo;, save.</li>
</ul>
<h3 id="investigate-the-small-plasmid-contig3">Investigate the small plasmid (contig3)</h3>
<ul>
<li>Blast the start of contig3 against itself</li>
<li>Take the start of the contig:</li>
</ul>
<pre><code>head -n 10 contig3.fa &gt; contig3.fa.head
</code></pre>

<ul>
<li>We want to see if it matches the end (overhang)</li>
<li>Format the assembly file for blast:</li>
</ul>
<pre><code>makeblastdb -in contig3.fa -dbtype nucl
</code></pre>

<ul>
<li>blast the start of the assembly (.head file) against all of the assembly:</li>
</ul>
<pre><code>blastn -query contig3.fa.head -db contig3.fa -evalue 1e-3 -dust no -out contig3.bls
</code></pre>

<ul>
<li>look at <fn>contig3.bls</fn> to see hits:</li>
</ul>
<pre><code>less contig3.bls
</code></pre>

<ul>
<li>the first hit is against itself, as expected</li>
<li>there are no few further hits, so we assume there is no overhang that needs trimming.</li>
<li>however, the sequence is likely then to be longer than this.</li>
</ul>
<pre><code>less contig3.fa
</code></pre>

<ul>
<li>Copy the sequence</li>
<li>Go to NCBI: <tt>https://blast.ncbi.nlm.nih.gov/Blast.cgi</tt>; choose blastx</li>
<li>Paste the sequence from contig3.fa</li>
<li>Choose genetic code = 11</li>
<li>Blast
<img alt="ncbi blast" src="./images/ncbi_blast_contig3.png" /></li>
<li>This hits a replication (plasmid) protein. Hypothesise that   this is a small plasmid; search for the entire sequence within the assembly of all the Illumina reads (next step).</li>
</ul>
<h3 id="assemble-all-the-illumina-reads">Assemble <em>all</em> the illumina reads</h3>
<ul>
<li>Assemble all the Illumina reads with spades (not just those reads that did not map to the pacbio assembly).</li>
</ul>
<pre><code>spades-fast --R1 R1.fastq.gz --R2 R2.fastq.gz --gsize 2.8M --outdir spades_fast --cpus 32
</code></pre>

<pre><code>cd spades_fast
</code></pre>

<ul>
<li>in here is the <fn>assembly_graph.fastg</fn></li>
<li>Transfer this file to your local computer (e.g. using the file transfer program <a href="https://cyberduck.io/?l=en">Cyberduck</a>).</li>
<li>Examine the assembly in the program <a href="https://rrwick.github.io/Bandage/">Bandage</a>.<ul>
<li>File: Load graph: <fn>assembly_graph.fastg</fn></li>
<li>In the left hand panel, click &ldquo;Draw graph&rdquo;</li>
<li>Your assembly graph may look like this:
<img alt="bandage pic" src="./images/illumina_assembly_bandage.png" /></li>
</ul>
</li>
<li>
<p>Blast the small plasmid sequence in this assembly</p>
<ul>
<li>In the left hand panel: Blast: create/view BLAST search</li>
<li>Build blast database</li>
<li>Paste in the sequence of contig3</li>
<li>Blast</li>
<li>The main hit is around node 10.</li>
</ul>
</li>
<li>
<p>Go to the main Bandage window</p>
<ul>
<li>&ldquo;find nodes&rdquo; in right hand panel - 10</li>
<li>This node is slightly longer: 2373: this could be the plasmid</li>
<li>Extract this node in fasta format: In the top panel, go to Output: Save selected node sequences; save as <fn>contig3b.fa</fn></li>
<li>Open this file in nano and change the header to &ldquo;&gt;contig3b&rdquo;, save.</li>
</ul>
</li>
</ul>
<h3 id="trim-small-plasmid">Trim small plasmid</h3>
<ul>
<li>Take the start of the sequence and see if it matches the end:</li>
</ul>
<pre><code>head -n 10 contig3b.fa &gt; contig3b.fa.head
makeblastdb -in contig3b.fa -dbtype nucl
blastn -query contig3b.fa.head -db contig3b.fa -evalue 1e-3 -dust no -out contig3b.bls
less contig3b.bls
</code></pre>

<ul>
<li>The first hit is against the start of the chromosome, as expected.</li>
<li>The last hit starts at position 2253; we will trim the plasmid to position 2252</li>
<li>Index the contig3b.fa file:</li>
</ul>
<pre><code>samtools faidx contig3b.fa
</code></pre>

<ul>
<li>Trim:</li>
</ul>
<pre><code>samtools faidx contig3b.fa contig3b:1-2252 &gt; contig3b.fa.trimmed
</code></pre>

<ul>
<li>Open this file in nano and change the header to &ldquo;&gt;contig3b&rdquo;, save.</li>
<li>We now have a trimmed contig3b.</li>
</ul>
<h3 id="collect-all-contigs-in-one-file">Collect all contigs in one file</h3>
<pre><code>cat contig_1_2.fa contig3b.fa.trimmed &gt; all_contigs.fa
</code></pre>

<ul>
<li>See the three contigs and sizes:</li>
</ul>
<pre><code>fa -f all_contigs.fa
</code></pre>

<h2 id="correct">Correct</h2>
<p>We will correct the pacbio assembly, first with pacbio corrected reads (from Canu) and then with Illumina reads.</p>
<ul>
<li>inputs:<ul>
<li>Draft pacbio assembly (overhang trimmed from each of the three replicons)</li>
<li>corrected pacbio reads from Canu.</li>
<li>illumina reads (aligned to pacbio assembly: in bam format)</li>
</ul>
</li>
<li>output: corrected assembly</li>
</ul>
<h3 id="1-correct-with-pacbio-corrected-reads">1. Correct with pacbio corrected reads</h3>
<p>Align reads to assembly:</p>
<pre><code class="bash">bwa index all_contigs.fa
bwa mem -t 32 all_contigs.fa canu.correctedReads.fasta.gz | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx all_contigs.fa
</code></pre>

<ul>
<li><strong>-t</strong> is the number of cores (e.g. 8)<ul>
<li>to find out how many you have, grep -c processor /proc/cpuinfo</li>
</ul>
</li>
<li>now we have an alignment file to use in pilon: <fn>aln.bam</fn></li>
</ul>
<p>Run pilon:</p>
<pre><code class="bash">pilon --genome all_contigs.fa --frags aln.bam --output pilon1 --fix all --mindepth 0.5 --changes --threads 32 --verbose
</code></pre>

<ul>
<li><strong>&ndash;genome</strong> is the name of the input assembly to be corrected</li>
<li><strong>&ndash;frags</strong> is the alignment of the reads against the assembly</li>
<li><strong>&ndash;output</strong> is the name of the output directory</li>
<li><strong>&ndash;fix</strong> is an option for types of corrections</li>
<li><strong>&ndash;mindepth</strong> gives a minimum read depth to use</li>
<li><strong>&ndash;changes</strong> produces an output file of the changes made</li>
<li><strong>&ndash;threads</strong> is the number of cores</li>
<li><strong>&ndash;verbose</strong> prints information to the screen during the run</li>
</ul>
<p>Look at the .changes file:</p>
<pre><code class="bash">less pilon1.changes | column -t
</code></pre>

<ul>
<li>This shows the corrections made by Pilon:</li>
</ul>
<p><img alt="Pilon" src="./images/pilon.png" /></p>
<ul>
<li>Look at the details of the .fasta file:</li>
</ul>
<pre><code>fa -f pilon1.fasta
</code></pre>

<h3 id="2-correct-with-illumina-reads">2. Correct with Illumina reads</h3>
<p>Input: the corrected assembly from the previous step: <fn>pilon1.fasta</fn></p>
<p>Align the Illumina reads to this assembly:</p>
<pre><code>bwa index pilon1.fasta
bwa mem -t 32 pilon1.fasta R1.fastq.gz R2.fastq.gz | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx pilon1.fasta
</code></pre>

<p>Look at how the illumina reads are aligned:</p>
<pre><code>samtools tview -p contig1 aln.bam pilon1.fasta
</code></pre>

<p>note: <strong>contig1</strong> is the name of the contig to view; e.g. tig00000000.</p>
<p>Run pilon:</p>
<pre><code class="bash">pilon --genome pilon1.fa --frags aln.bam --output pilon2 --fix all --mindepth 0.5 --changes --threads 32 --verbose
</code></pre>

<p>Look at the changes file:</p>
<pre><code class="bash">less pilon1.changes
</code></pre>

<p>Look at the .fasta file:</p>
<pre><code>less pilon2.fasta
</code></pre>

<p>If there are more than 2 changes, run Pilon again, using the pilon2.fasta file as the input assembly, and the Illumina reads to correct.</p>
<p><strong>Final output:</strong></p>
<ul>
<li>the corrected genome assembly of <em>Staphylococcus aureus</em> in .fasta format, containing three contigs: chromosome, large plasmid and small plasmid.  </li>
</ul>
<h2 id="next">Next</h2>
<p><strong>Further analyses:</strong></p>
<ul>
<li>Annotate with Prokka.</li>
<li>Comparative genomics, e.g. with Roary.</li>
</ul>
<p><strong>Links:</strong></p>
<ul>
<li><a href="https://s3.amazonaws.com/files.pacb.com/software/instrument/2.0.0/bas.h5+Reference+Guide.pdf">Details of bas.h5 files</a></li>
<li>Canu <a href="http://canu.readthedocs.io/en/stable/quick-start.html">manual</a> and <a href="https://github.com/marbl/canu">gitub repository</a></li>
<li>Circlator <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0849-0">article</a> and <a href="http://sanger-pathogens.github.io/circlator/">github repository</a></li>
<li>Pilon <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0112963">article</a> and <a href="https://github.com/broadinstitute/pilon/wiki">github repository</a></li>
<li>Notes on <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Finishing-Bacterial-Genomes">finishing</a> and <a href="https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Evaluating-Assemblies">evaluating</a> assemblies.</li>
</ul>
<h2 id="commands-summary">Commands summary <a name="summary"></a></h2>
<pre><code class="bash">//get the sequencing reads
//joined pacbio reads: subreads.fastq
//illumina reads: R1.fastq.gz, R2.fastq.gz

//run canu to assemble
canu -p staph -d output genomeSize=2.8m -pacbio-raw subreads.fastq
//output: staph.contigs.fasta

//run circlator to trim and orient
circlator all --verbose ../staph.contigs.fasta ../staph.corrected.reads.fastq.gz circlator_all_output
//output: 06.fixstart.fasta; rename as contig_1_2.fa

//find smaller plasmids

//align illumina to pacbio contigs
bwa index contig_1_2.fa
bwa mem -t 8 contig_1_2.fa R1.fastq.gz R2.fastq.gz | samtools sort &gt; aln.bam
//output is aln.bam

//extract unmapped reads
samtools index aln.bam
samtools fastq -f 4 -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq aln.bam

//assemble these reads with spades
spades.py -1 unmapped.R1.fastq -2 unmapped.R2.fastq -s unmapped.RS.fastq -o spades_assembly
//save any long contigs; e.g. as contig3.fa

//blast this contig to find overhang
head -n 10 contig3.fa &gt; contig3.fa.head
makeblastdb -in contig3.fa -dbtype nucl
blastn -query contig3.fa.head -db contig3.fa -evalue 1e-3 -dust no -out contig3.bls
less contig3.bls
//no overhang found

//assemble all illumina reads
spades-fast --R1 R1.fastq.gz --R2 R2.fastq.gz --gsize 2.8M --outdir spades_fast --cpus 32
//blast the assembly graph with contig3.fa; extract out node and save as contig3b.fa

//trim overhang
head -n 10 contig3b.fa &gt; contig3b.fa.head
makeblastdb -in contig3b.fa -dbtype nucl
blastn -query contig3b.fa.head -db contig3b.fa -evalue 1e-3 -dust no -out contig3b.bls
less contig3b.bls
samtools faidx contig3b.fa
samtools faidx contig3b.fa contig3b:1-2252 &gt; contig3b.fa.trimmed
//open contig3b.fa in nano and shorten header name

//join all contigs: all_contigs.fa

//correct with pilon

//1. correct using corrected pacbio reads
bwa index all_contigs.fa
bwa mem -t 32 all_contigs.fa canu.correctedReads.fasta.gz | samtools sort &gt; aln.bam
samtools index aln.bam
samtools faidx all_contigs.fa
pilon --genome all_contigs.fa --frags aln.bam --output corrected --fix bases --mindepth 0.5 --changes --threads 32 --verbose

//2. correct using illumina reads - use output from 1 as the contigs file

//3. repeat correction using Illumina reads - use output from 2 as the contigs file

//output: corrected genome assembly of Staphylococcus aureus in .fasta format, containing three contigs: chromosome, large plasmid and small plasmid.
</code></pre></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>