<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Pathway annotation and prediction - ABRPI Training 2017</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics Workshop <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../workshop_overview/">Workshop overview</a>
</li>

                    
                        
<li >
    <a href="../../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../../xtandem/">Protein Identification</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Pathway Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li class="active">
    <a href="./">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../../xtandem/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../../about/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#pathway-prediction-and-annotations-for-new-organisms">Pathway prediction and annotations for new organisms</a></li>
        
            <li><a href="#connect-to-mgvl-using-vnc">Connect to mGVL using VNC</a></li>
        
            <li><a href="#create-a-new-database">Create a new Database</a></li>
        
            <li><a href="#predict-and-annotate-the-pathways">Predict and annotate the pathways</a></li>
        
            <li><a href="#verify-and-test-the-new-database">Verify and test the new Database</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="pathway-prediction-and-annotations-for-new-organisms">Pathway prediction and annotations for new organisms</h1>
<h2 id="connect-to-mgvl-using-vnc">Connect to mGVL using VNC</h2>
<ul>
<li>Go to the mGVL dashboard</li>
<li>Click on the link next to the <strong>Lubuntu Destkop</strong> (http://your-mgvl-ip-address/vnc)</li>
</ul>
<p><img alt="mGVL-dashboard" src="./images/mGVL-dashboard.png" /></p>
<p>A new browser will appear, enter your user credentials to login</p>
<p><img alt="VNC-login" src="./images/VNC-login.png" /></p>
<h2 id="create-a-new-database">Create a new Database</h2>
<p>Once logged in, you should see the following desktop with the 3 Pathway tools icons on the left. Click on the first
icon <strong>Pathway Tools v1.9.5</strong>. This will bring up a new window.</p>
<p><img alt="pathwayTools-icons" src="./images/pathwayTools-icons.png" /></p>
<ul>
<li>In the <strong>Pathway Tools</strong> window, click on <ss>Tools</ss> menu item and then click on <ss>PathLogic</ss></li>
<li>In the new window, at the top, click on <ss>Database</ss> and then <ss>Create New</ss></li>
</ul>
<p>Provide the metadata for the new database. In the <strong>Database(required)</strong> section, enter the following values:</p>
<ul>
<li><strong>Organism/Project ID:</strong> <ss>SEPSIS25707</ss></li>
<li><strong>Database Name:</strong> <ss>SepsisCycl</ss></li>
</ul>
<p>Leave the other parameters as the default values, that is:
<em>   Version: 1.0 (default)
</em>   DB Storage Type: File (default)</p>
<p><img alt="meta-database" src="./images/meta-database.png" /></p>
<p>Under the <strong>Taxonomy (required)</strong> section:</p>
<ul>
<li>Check Box if this is a multi-organism database: <ss>uncheck</ss></li>
<li>In <strong>Organism taxonomic class</strong>, type &ldquo;1314&rdquo; and click on <ss>Select</ss>. The species name of Streptococcus pyogenes will appear and a popup window will also appear. Click on to close the window.</li>
<li>Create organism?: click <ss>yes</ss></li>
<li>Strain: <ss>HKU419(ARP)</ss> for sample (25707)</li>
<li>Genome Source: <ss>ARP</ss></li>
<li>NCBI taxonomy ID: <ss>1314</ss> (come back to this after Select &ldquo;Strain&rdquo; in next step)</li>
<li>Rank : Select <ss>Strain</ss></li>
</ul>
<p>Leave all other fields as their default values:</p>
<ul>
<li>Full Species Name: autocomplete from the previous step</li>
<li>Abbreviated Species Name: auto complete from the previous step</li>
<li>Subspecies : leave it blank</li>
<li>Default Codon Table: 11 - Bacterial and Plant Plastid&hellip;</li>
<li>Mitochondrial Codon Table : 0 - Unspecified</li>
</ul>
<p><img alt="meta-taxonomy" src="./images/meta-taxonomy.png" /></p>
<p>Leave <strong>Credits(optional)</strong> section black.</p>
<ul>
<li>Click <ss>OK</ss>.</li>
</ul>
<p>Another window will appear while processing. Wait for processing to complete, until you see the next window. Click on <ss>Enter Replicon Editor</ss>.</p>
<p><img alt="processingComplete" src="./images/processingComplete.png" /></p>
<h3 id="specify-replicon-details">Specify Replicon details</h3>
<p>In this view, you provide the details of the annotated assembly. Each chromosome will have a separate entry. For each chromosome, you need to provide the <strong>GBK</strong> and <strong>FNA</strong> annotation files from Prooka.</p>
<ul>
<li>Name : <ss>1</ss> ( This is the chromosome name or number)</li>
<li>Circular: <ss>checked</ss> if circulator was performed or leave it unchecked</li>
<li>Select annotation file: select the gbk from prokka</li>
<li>Select sequence file: select the fna from prokka</li>
</ul>
<p>Leave other fields as their default value:</p>
<ul>
<li>Type: Chromosome (default)</li>
<li>Code: Bacterial, Archaeal and Plant Plastid (default)</li>
<li>ID: leave it blank (default)</li>
<li>Links to other database: NCBI Reference Sequences Database (default)</li>
<li>
<p>Relationship : same Entity (default)</p>
</li>
<li>
<p>Click <ss>OK</ss></p>
</li>
</ul>
<p><img alt="specify-replicons" src="./images/specify-replicons.png" /></p>
<p>The metada for this database has now been created. Next we need to predict the pathways and annotate them.</p>
<h2 id="predict-and-annotate-the-pathways">Predict and annotate the pathways</h2>
<p>Still on the previous window, from the top menu bar, click on <ss>Build</ss> and then <ss>Automated Build</ss>. This will take a while depending on the number of replicons that was included for this database.</p>
<p><img alt="build-autobuild" src="./images/build-autobuild.png" /></p>
<p>Once the process has completed, a new window, <strong>Pathway Scoring Parameters</strong> will appear. Leave the values as their default values.</p>
<ul>
<li>Taxonomic Pruning: Enabled (default)</li>
<li>Pathway Prediction Score Cutoff: 0.15 (default).</li>
</ul>
<p>A higher cutoff value for the <em>prediction score</em> will mean less pathways are predicted. This is the level of stringency imposed on the prediction.</p>
<p>Click <ss>OK</ss>.</p>
<p><img alt="pathway-scoring" src="./images/pathway-scoring.png" /></p>
<p>Again the process can take several minutes depending on the number of replicons included. Wait until the process is complete.</p>
<p>Click on <ss>Database</ss> and then <ss>Save DB</ss> from the menu. The pathways have now been predicted, annotated and stored in the database.  </p>
<h2 id="verify-and-test-the-new-database">Verify and test the new Database</h2>
<p>Now that the database has been created, we need to verify that it is available for use. First we check that it is present in the Pathway Tools. On the main screen, <strong>Pathway Tools - Available Databases</strong>, you should see a new entry at the bottom of the pre-existing list, <strong>&ldquo;Streptococcus pyogenes HKU419(ARP)&rdquo;</strong>.</p>
<p><img alt="new-database" src="./images/new-database.png" /></p>
<p>To see this newly created database in the Pathway Tools web-application, start up the Pathway Tools Web service (if not already running).</p>
<ul>
<li>You can close all the windows, to exit Pathway Tools</li>
<li>If the web server is not already running, click on the <ss>Pathway Tool Web server</ss> from the desktop, this is the third icon.</li>
<li>Open an internet brwoser and go to http://your-mgvl-ip-address:1555</li>
</ul>
<p><img alt="pathway-tools-site" src="./images/pathwaytools-website.png" /></p>
<h3 id="using-the-new-database-in-pathway-tools-web-service">Using the new database in Pathway Tools Web service</h3>
<ul>
<li>On the main screen, click on <ss>change organism database</ss> at the top right corner, under the search box.</li>
<li>A new window will appear, select the newly created species from the list</li>
<li>Click <ss>OK</ss></li>
</ul>
<p><img alt="pathway-change-database" src="./images/changeDatabase.png" />
<img alt="pathway-select-database" src="./images/selectDatabase.png" /></p>
<p>A statistic table of the species will be available for overview.</p>
<p><img alt="statTable" src="./images/statTable.png" /></p>
<p>Click on the <ss>Pathways</ss> link in the table, which will show the list of all predicted pathways in this database.</p>
<p><img alt="pathway-list" src="./images/pathwayList.png" /></p>
<p>You can navigate through the list of pathways and highlight a pathway of interest. For example,</p>
<ul>
<li>Click on the <ss>+</ss> symbol next to <strong>Detoxification</strong></li>
<li>Click on the <ss>+</ss> symbol again next to <strong>Antibiotic Resistance</strong></li>
<li>Click on <ss>peptidoglycan biosynthesis V (beta-lactam resistance)</ss> which will bring up the pathway view  </li>
</ul>
<p><img alt="antibioticResistancePathway" src="./images/antibioticResistancePathway.png" /></p>
<p>Go back to the previous screen with the statistics summary table and this time click on the <ss>1</ss> under the <strong>Replicon</strong> heading in the top table. The following genome browser will be shown.</p>
<p><img alt="genomeBrowser" src="./images/genomeBrowser.png" /></p>
<p>From here forward the user can explore using the Pathway Tools webservice.</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script>var base_url = '../../..';</script>
        <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
        <script src="../../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>