
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Tutorials for analysing bacterial omics data">
      
      
        <link rel="canonical" href="http://sepsis-omics.github.io/tutorials/modules/pathway_tools/annotation/">
      
      
        <meta name="author" content="Anna Syme, Torsten Seemann">
      
      
        <link rel="shortcut icon" href="../../../favicon.ico">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.6.1">
    
    
      
        <title>Pathway annotation and prediction - ABRPI</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-4d0d3f2fbf.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/sepsis.css">
    
    
  </head>
  
  
  
  
    <body>
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="http://sepsis-omics.github.io/tutorials/" title="ABRPI" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Proteomics
                </span>
              
            
            Pathway annotation and prediction
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
            


  


  <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../workshop_overview/" title="Genomics" class="md-tabs__link">
          Genomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../dge/" title="Transcriptomics" class="md-tabs__link">
          Transcriptomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../xtandem/" title="Proteomics" class="md-tabs__link md-tabs__link--active">
          Proteomics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../xcms/" title="Metabolomics" class="md-tabs__link">
          Metabolomics
        </a>
      
    </li>
  

      
        
      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    ABRPI
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/sepsis-omics/tutorials" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      sepsis-omics/tutorials
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Genomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Genomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../workshop_overview/" title="Genomics workshop overview" class="md-nav__link">
      Genomics workshop overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../galaxy/" title="Starting with Galaxy" class="md-nav__link">
      Starting with Galaxy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../data-dna/" title="Training dataset" class="md-nav__link">
      Training dataset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../fastqc/" title="Quality control" class="md-nav__link">
      Quality control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../velvet/" title="Genome assembly with Velvet" class="md-nav__link">
      Genome assembly with Velvet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../spades/" title="Genome assembly with Spades" class="md-nav__link">
      Genome assembly with Spades
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../prokka/" title="Genome annotation" class="md-nav__link">
      Genome annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../snippy/" title="Variant finding" class="md-nav__link">
      Variant finding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../access_data/" title="Public data &rarr; assembly, annotation" class="md-nav__link">
      Public data &rarr; assembly, annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../abricate/" title="Finding antibiotic-resistant genes" class="md-nav__link">
      Finding antibiotic-resistant genes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../viral_genomes/" title="Viral genome sequencing" class="md-nav__link">
      Viral genome sequencing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../workflows/" title="Galaxy workflows" class="md-nav__link">
      Galaxy workflows
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../frogs/" title="Metagenomics with FROGS" class="md-nav__link">
      Metagenomics with FROGS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../kraken/" title="Sample identification with Kraken" class="md-nav__link">
      Sample identification with Kraken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../pacbio/" title="PacBio assembly with SMRT portal" class="md-nav__link">
      PacBio assembly with SMRT portal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cmdline_assembly/" title="PacBio assembly with command line tools" class="md-nav__link">
      PacBio assembly with command line tools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../spades_cmdline/" title="Spades assembly - commandline" class="md-nav__link">
      Spades assembly - commandline
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../roary/" title="Pangenomes with Roary" class="md-nav__link">
      Pangenomes with Roary
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Transcriptomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Transcriptomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../dge/" title="Differential gene expression using Galaxy and Degust" class="md-nav__link">
      Differential gene expression using Galaxy and Degust
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../kallisto/" title="Differential gene expression using Kallisto and Degust" class="md-nav__link">
      Differential gene expression using Kallisto and Degust
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked>
    
    <label class="md-nav__link" for="nav-4">
      Proteomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Proteomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../xtandem/" title="Protein Identification" class="md-nav__link">
      Protein Identification
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Pathway annotation and prediction
      </label>
    
    <a href="./" title="Pathway annotation and prediction" class="md-nav__link md-nav__link--active">
      Pathway annotation and prediction
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#connect-to-mgvl-using-vnc" title="Connect to mGVL using VNC" class="md-nav__link">
    Connect to mGVL using VNC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-a-new-database" title="Create a new Database" class="md-nav__link">
    Create a new Database
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#specify-replicon-details" title="Specify Replicon details" class="md-nav__link">
    Specify Replicon details
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#predict-and-annotate-the-pathways" title="Predict and annotate the pathways" class="md-nav__link">
    Predict and annotate the pathways
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#verify-and-test-the-new-database" title="Verify and test the new Database" class="md-nav__link">
    Verify and test the new Database
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-new-database-in-pathway-tools-web-service" title="Using the new database in Pathway Tools Web service" class="md-nav__link">
    Using the new database in Pathway Tools Web service
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Metabolomics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metabolomics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../xcms/" title="Metabolite identification" class="md-nav__link">
      Metabolite identification
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../about/" title="About" class="md-nav__link">
      About
    </a>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#connect-to-mgvl-using-vnc" title="Connect to mGVL using VNC" class="md-nav__link">
    Connect to mGVL using VNC
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-a-new-database" title="Create a new Database" class="md-nav__link">
    Create a new Database
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#specify-replicon-details" title="Specify Replicon details" class="md-nav__link">
    Specify Replicon details
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#predict-and-annotate-the-pathways" title="Predict and annotate the pathways" class="md-nav__link">
    Predict and annotate the pathways
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#verify-and-test-the-new-database" title="Verify and test the new Database" class="md-nav__link">
    Verify and test the new Database
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#using-the-new-database-in-pathway-tools-web-service" title="Using the new database in Pathway Tools Web service" class="md-nav__link">
    Using the new database in Pathway Tools Web service
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <p><br></p>
<h1 id="pathway-prediction-and-annotations-for-new-organisms">Pathway prediction and annotations for new organisms</h1>
<h2 id="connect-to-mgvl-using-vnc">Connect to mGVL using VNC</h2>
<ul>
<li>Go to the mGVL dashboard</li>
<li>Click on the link next to the <strong>Lubuntu Destkop</strong> (http://your-mgvl-ip-address/vnc)</li>
</ul>
<p><img alt="mGVL-dashboard" src="./images/mGVL-dashboard.png" /></p>
<p>A new browser will appear, enter your user credentials to login</p>
<p><img alt="VNC-login" src="./images/VNC-login.png" /></p>
<h2 id="create-a-new-database">Create a new Database</h2>
<p>Once logged in, you should see the following desktop with the 3 Pathway tools icons on the left. Click on the first
icon <strong>Pathway Tools v1.9.5</strong>. This will bring up a new window.</p>
<p><img alt="pathwayTools-icons" src="./images/pathwayTools-icons.png" /></p>
<ul>
<li>In the <strong>Pathway Tools</strong> window, click on <ss>Tools</ss> menu item and then click on <ss>PathLogic</ss></li>
<li>In the new window, at the top, click on <ss>Database</ss> and then <ss>Create New</ss></li>
</ul>
<p>Provide the metadata for the new database. In the <strong>Database(required)</strong> section, enter the following values:</p>
<ul>
<li><strong>Organism/Project ID:</strong> <ss>SEPSIS25707</ss></li>
<li><strong>Database Name:</strong> <ss>SepsisCycl</ss></li>
</ul>
<p>Leave the other parameters as the default values, that is:
<em>   Version: 1.0 (default)
</em>   DB Storage Type: File (default)</p>
<p><img alt="meta-database" src="./images/meta-database.png" /></p>
<p>Under the <strong>Taxonomy (required)</strong> section:</p>
<ul>
<li>Check Box if this is a multi-organism database: <ss>uncheck</ss></li>
<li>In <strong>Organism taxonomic class</strong>, type &ldquo;1314&rdquo; and click on <ss>Select</ss>. The species name of Streptococcus pyogenes will appear and a popup window will also appear. Click on to close the window.</li>
<li>Create organism?: click <ss>yes</ss></li>
<li>Strain: <ss>HKU419(ARP)</ss> for sample (25707)</li>
<li>Genome Source: <ss>ARP</ss></li>
<li>NCBI taxonomy ID: <ss>1314</ss> (come back to this after Select &ldquo;Strain&rdquo; in next step)</li>
<li>Rank : Select <ss>Strain</ss></li>
</ul>
<p>Leave all other fields as their default values:</p>
<ul>
<li>Full Species Name: autocomplete from the previous step</li>
<li>Abbreviated Species Name: auto complete from the previous step</li>
<li>Subspecies : leave it blank</li>
<li>Default Codon Table: 11 - Bacterial and Plant Plastid&hellip;</li>
<li>Mitochondrial Codon Table : 0 - Unspecified</li>
</ul>
<p><img alt="meta-taxonomy" src="./images/meta-taxonomy.png" /></p>
<p>Leave <strong>Credits(optional)</strong> section black.</p>
<ul>
<li>Click <ss>OK</ss>.</li>
</ul>
<p>Another window will appear while processing. Wait for processing to complete, until you see the next window. Click on <ss>Enter Replicon Editor</ss>.</p>
<p><img alt="processingComplete" src="./images/processingComplete.png" /></p>
<h3 id="specify-replicon-details">Specify Replicon details</h3>
<p>In this view, you provide the details of the annotated assembly. Each chromosome will have a separate entry. For each chromosome, you need to provide the <strong>GBK</strong> and <strong>FNA</strong> annotation files from Prokka.</p>
<ul>
<li>Name : <ss>1</ss> ( This is the chromosome name or number)</li>
<li>Circular: <ss>checked</ss> if circulator was performed or leave it unchecked</li>
<li>Select annotation file: select the gbk from prokka</li>
<li>Select sequence file: select the fna from prokka</li>
</ul>
<p>Leave other fields as their default value:</p>
<ul>
<li>Type: Chromosome (default)</li>
<li>Code: Bacterial, Archaeal and Plant Plastid (default)</li>
<li>ID: leave it blank (default)</li>
<li>Links to other database: NCBI Reference Sequences Database (default)</li>
<li>
<p>Relationship : same Entity (default)</p>
</li>
<li>
<p>Click <ss>OK</ss></p>
</li>
</ul>
<p><img alt="specify-replicons" src="./images/specify-replicons.png" /></p>
<p>The metada for this database has now been created. Next we need to predict the pathways and annotate them.</p>
<h2 id="predict-and-annotate-the-pathways">Predict and annotate the pathways</h2>
<p>Still on the previous window, from the top menu bar, click on <ss>Build</ss> and then <ss>Automated Build</ss>. This will take a while depending on the number of replicons that was included for this database.</p>
<p><img alt="build-autobuild" src="./images/build-autobuild.png" /></p>
<p>Once the process has completed, a new window, <strong>Pathway Scoring Parameters</strong> will appear. Leave the values as their default values.</p>
<ul>
<li>Taxonomic Pruning: Enabled (default)</li>
<li>Pathway Prediction Score Cutoff: 0.15 (default).</li>
</ul>
<p>A higher cutoff value for the <em>prediction score</em> will mean less pathways are predicted. This is the level of stringency imposed on the prediction.</p>
<p>Click <ss>OK</ss>.</p>
<p><img alt="pathway-scoring" src="./images/pathway-scoring.png" /></p>
<p>Again the process can take several minutes depending on the number of replicons included. Wait until the process is complete.</p>
<p>Click on <ss>Database</ss> and then <ss>Save DB</ss> from the menu. The pathways have now been predicted, annotated and stored in the database.  </p>
<h2 id="verify-and-test-the-new-database">Verify and test the new Database</h2>
<p>Now that the database has been created, we need to verify that it is available for use. First we check that it is present in the Pathway Tools. On the main screen, <strong>Pathway Tools - Available Databases</strong>, you should see a new entry at the bottom of the pre-existing list, <strong>&ldquo;Streptococcus pyogenes HKU419(ARP)&rdquo;</strong>.</p>
<p><img alt="new-database" src="./images/new-database.png" /></p>
<p>To see this newly created database in the Pathway Tools web-application, start up the Pathway Tools Web service (if not already running).</p>
<ul>
<li>You can close all the windows, to exit Pathway Tools</li>
<li>If the web server is not already running, click on the <ss>Pathway Tool Web server</ss> from the desktop, this is the third icon.</li>
<li>Open an internet brwoser and go to http://your-mgvl-ip-address:1555</li>
</ul>
<p><img alt="pathway-tools-site" src="./images/pathwaytools-website.png" /></p>
<h3 id="using-the-new-database-in-pathway-tools-web-service">Using the new database in Pathway Tools Web service</h3>
<ul>
<li>On the main screen, click on <ss>change organism database</ss> at the top right corner, under the search box.</li>
<li>A new window will appear, select the newly created species from the list</li>
<li>Click <ss>OK</ss></li>
</ul>
<p><img alt="pathway-change-database" src="./images/changeDatabase.png" />
<img alt="pathway-select-database" src="./images/selectDatabase.png" /></p>
<p>A statistic table of the species will be available for overview.</p>
<p><img alt="statTable" src="./images/statTable.png" /></p>
<p>Click on the <ss>Pathways</ss> link in the table, which will show the list of all predicted pathways in this database.</p>
<p><img alt="pathway-list" src="./images/pathwayList.png" /></p>
<p>You can navigate through the list of pathways and highlight a pathway of interest. For example,</p>
<ul>
<li>Click on the <ss>+</ss> symbol next to <strong>Detoxification</strong></li>
<li>Click on the <ss>+</ss> symbol again next to <strong>Antibiotic Resistance</strong></li>
<li>Click on <ss>peptidoglycan biosynthesis V (beta-lactam resistance)</ss> which will bring up the pathway view  </li>
</ul>
<p><img alt="antibioticResistancePathway" src="./images/antibioticResistancePathway.png" /></p>
<p>Go back to the previous screen with the statistics summary table and this time click on the <ss>1</ss> under the <strong>Replicon</strong> heading in the top table. The following genome browser will be shown.</p>
<p><img alt="genomeBrowser" src="./images/genomeBrowser.png" /></p>
<p>From here forward the user can explore using the Pathway Tools webservice.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../xtandem/" title="Protein Identification" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Protein Identification
              </span>
            </div>
          </a>
        
        
          <a href="../../xcms/" title="Metabolite identification" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metabolite identification
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2016-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://omics.data.edu.au/" class="md-footer-social__link fa fa-coffee"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials" class="md-footer-social__link fa fa-github"></a>
    
      <a href="https://github.com/sepsis-omics/tutorials/issues" class="md-footer-social__link fa fa-bug"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-6b599127bc.js"></script>
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>