<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Training dataset - ABRPI Training 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics Workshop <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../workshop_overview/">Workshop overview</a>
</li>

                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li class="active">
    <a href="./">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../frogs/">Metagenomics</a>
</li>

                    
                        
<li >
    <a href="../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xtandem/">Protein Identification</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Pathway Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../pathway_tools/annotation/">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../galaxy/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../fastqc/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#dataset">Dataset</a></li>
        
            <li><a href="#galaxy-histories-urls">Galaxy histories: URLs</a></li>
        
            <li><a href="#individual-input-files">Individual input files</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="dataset">Dataset</h1>
<p>This page contains data for the tutorials. The tutorials will specify whether to import a whole history (URLs listed in the first section) or an individual file (URLs listed in the second section).</p>
<h2 id="galaxy-histories-urls">Galaxy histories: URLs</h2>
<ul>
<li>Galaxy history of input files</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Galaxy_history_input_files.tar.gz</tt></p>
<ul>
<li>Galaxy history: FastQC</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/FastQChistory.tar.gz</tt></p>
<ul>
<li>Galaxy history: Spades</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Spadeshistory.tar.gz</tt></p>
<ul>
<li>Galaxy history: Prokka</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Prokkahistory.tar.gz</tt></p>
<ul>
<li>Galaxy history: Snippy</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Snippyhistory.tar.gz</tt></p>
<p><strong>To get the saved tutorial history (a set of files) into Galaxy:</strong></p>
<ul>
<li>Copy the link address.</li>
<li>Go to your Galaxy instance. Make sure you are registered and logged in. Refresh the page.</li>
<li>Click on the <ss>History</ss> cog <img alt="cog icon" src="./images/image02.png" /></li>
<li>Select <ss>Import from File</ss></li>
</ul>
<p><img alt="history options" src="./images/image03.png" /></p>
<ul>
<li>In the box called <ss>Archived History URL</ss>, paste in the link address to the Galaxy history.</li>
<li>Click <ss>Submit</ss></li>
<li>Wait a few seconds.</li>
<li>Click on the &ldquo;view all histories&rdquo; button <img alt="histories icon" src="./images/image11.png" /></li>
<li>See if the Galaxy history has been imported: it will be called <fn>imported from archive: Data</fn></li>
<li>Above that pane, click on the <ss>Switch to</ss> button.</li>
<li>Then click <ss>Done</ss> (in the top left corner).</li>
<li>You should now have a list of five files in your current history.</li>
</ul>
<p><img alt="files in galaxy history" src="./images/datafiles.png" /></p>
<h2 id="individual-input-files">Individual input files</h2>
<h3 id="wildtype-reference">Wildtype reference</h3>
<ul>
<li>wildtype.fna</li>
</ul>
<p><tt> https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/wildtype.fna</tt></p>
<ul>
<li>wildtype.gbk</li>
</ul>
<p><tt> https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/wildtype.gbk</tt></p>
<ul>
<li>wildtype.gff</li>
</ul>
<p><tt> https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/wildtype.gff</tt></p>
<h3 id="mutant-illumina-sequence">Mutant Illumina sequence</h3>
<ul>
<li>mutant_R1.fastq.gz</li>
</ul>
<p><tt> https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/mutant_R1.fastq.gz</tt></p>
<ul>
<li>mutant_R2.fastq.gz</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/mutant_R2.fastq.gz</tt></p>
<h3 id="assembled-contigs">Assembled contigs</h3>
<ul>
<li>SPAdes_contigs.fasta</li>
</ul>
<p><tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/SPAdes_contigs.fasta</tt></p>
<h3 id="upload-to-galaxy">Upload to Galaxy</h3>
<p>There are two ways to upload these files to Galaxy. You can either download to your local computer and upload to Galaxy, or you can tell Galaxy to directly upload the file from an external source.</p>
<p>Download and upload:</p>
<ul>
<li>Download required file(s) to your computer.</li>
<li>From the Galaxy tool panel, click on <ss>Get Data &rarr; Upload File</ss>  </li>
<li>Click the <ss>Choose local file</ss> button  </li>
<li>Find and select the <fn>file</fn> you downloaded and click <ss>Open</ss>  </li>
<li>Set the <ss>Type</ss> correctly.  </li>
<li>Click the <ss>Start</ss> button.  </li>
<li>Once the progress bar reaches 100%, click the <ss>Close</ss> button  </li>
<li>The file will now upload to your current history.</li>
</ul>
<p>Or, tell Galaxy to find the file from an external source:</p>
<ul>
<li>From the Galaxy tool panel, click on <ss>Get Data &rarr; Upload File</ss>  </li>
<li>Click the <ss>Paste/Fetch data</ss> button  </li>
<li>Paste the URL into the box.</li>
<li>Click the <ss>Start</ss> button.  </li>
<li>Once the progress bar reaches 100%, click the <ss>Close</ss> button  </li>
<li>The file will now upload to your current history.</li>
</ul>
<!-- ## What next?

- If you are are working through the tutorials in a different order or want to see the completed history for another section, additional Galaxy histories are available above.
- Next: [Learn about quality control](../fastqc/index.md).
--></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>