<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Genome assembly with Velvet - ABRPI Training 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../workshop_overview/">Genomics workshop overview</a>
</li>

                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li class="active">
    <a href="./">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../frogs/">Metagenomics</a>
</li>

                    
                        
<li >
    <a href="../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                        
<li >
    <a href="../kallisto/">Differential gene expression using Kallisto and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics and Pathways <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xtandem/">Protein Identification</a>
</li>

                    
                        
<li >
    <a href="../pathway_tools/annotation/">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Metabolomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xcms/">Metabolite identification</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../fastqc/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../spades/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#assembly-using-velvet">Assembly using Velvet</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning objectives</a></li>
        
            <li><a href="#import-and-view-data">Import and view data</a></li>
        
            <li><a href="#assemble-reads-with-velvet">Assemble reads with Velvet</a></li>
        
            <li><a href="#examine-the-output">Examine the output</a></li>
        
            <li><a href="#assembly-with-velvet-optimiser">Assembly with Velvet Optimiser</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="assembly-using-velvet">Assembly using Velvet</h1>
<p>Keywords: de novo assembly, Velvet, Galaxy, Microbial Genomics Virtual Lab</p>
<h2 id="background">Background</h2>
<p>Velvet is one of a number of <em>de novo</em> assemblers that use short read sets as input (e.g. Illumina Reads), and the assembly method is based on de Bruijn graphs. For information about Velvet see this <a href="https://en.wikipedia.org/wiki/Velvet_assembler">link</a>.</p>
<!---
A protocol for assembling with Velvet (another *de novo* assembler) is available [here](https://docs.google.com/document/d/1xs-TI5MejQARqo0pcocGlymsXldwJbJII890gnmjI0o/pub).
--->

<p>In this activity, we will perform a <em>de novo</em> assembly of a short read set using the Velvet assembler.</p>
<h2 id="learning-objectives">Learning objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>assemble the reads using Velvet, and</li>
<li>examine the output assembly.</li>
</ol>
<h2 id="import-and-view-data">Import and view data</h2>
<p>If you have completed the previous tutorial on <a href="../fastqc/">Quality Control</a>, you should already have the required files in your current Galaxy history. If not, see how to get them <a href="../data-dna/">here</a>.</p>
<!-- Alternative ways to obtain the files:

- In the centre of the top Galaxy panel, select <ss>Shared Data: Histories</ss>
- Select <fn>Microbial-assembly-start-history</fn> and then in the top right corner click on <ss>Import history</ss>.
- This should now be your current history (the right hand panel) with five files.

Or:

- Click on the <ss>History</ss> cog ![cog icon](images/image02.png)
- Select <ss>Import from File</ss>
- In the box called <ss>Archived History URL</ss>, paste in this link address to the Galaxy history of input files:

<tt>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Galaxy_history_input_files.tar.gz</tt>

- Click <ss>Submit</ss>
- Wait a few seconds.
- Click on the "view all histories" button ![histories icon](images/histories.png)
- See if the Galaxy history has been imported: it will be called <fn>imported from archive: Data</fn>
- Above that pane, click on the <ss>Switch to</ss> button.
- Then click <ss>Done</ss> (in the top left corner).
- You should now have a list of five files in your current history.
-->

<h3 id="the-data">The data</h3>
<p>The read set for today is from an imaginary <em>Staphylococcus aureus</em> bacterium with a miniature genome. The whole genome shotgun method used to sequence our mutant strain read set was produced on an Illumina DNA sequencing instrument.</p>
<ul>
<li>The files we need for assembly are the <fn>mutant_R1.fastq</fn> and <fn>mutant_R2.fastq</fn>.</li>
<li>
<p>(We don&rsquo;t need the reference genome sequences for this tutorial).</p>
</li>
<li>
<p>The reads are paired-end.</p>
</li>
<li>
<p>Each read is 150 bases long. <!--(before trimming)--></p>
</li>
<li>
<p>The number of bases sequenced is equivalent to 19x the genome sequence of the wildtype strain. (Read coverage 19x - rather low!).</p>
</li>
</ul>
<!--
- <fn>wildtype.fna</fn>: the reference genome sequence of the wildtype strain in fasta format (a header line, then the nucleotide sequence of the genome)

- <fn>wildtype.gff</fn>: the reference genome sequence of the wildtype strain in general feature format (a list of features - one feature per line, then the nucleotide sequence of the genome).

- <fn>wildtype.gbk</fn>: the reference genome sequence in genbank format.
--->

<ul>
<li>Click on the View Data button (the <img alt="Eye icon" src="./images/image04.png" />) next to one of the FASTQ sequence files.</li>
</ul>
<!--
- The gff file should look like this:
- Brief Discussion about the GFF format (FIXME: add)
![GFF format](images/image08.png)

## Evaluate the input reads

Questions you might ask about your input reads include:

- How good is my read set?
- Do I need to ask for a new sequencing run?  
- Is it suitable for the analysis I need to do?

We will evaluate the input reads using the FastQC tool.

- This runs a standard series of tests on your read set and returns a relatively easy-to-interpret report.
- We will use the FastQC tool in Galaxy to evaluate the quality of one of our FASTQ files.
- Go to <ss>Tools &rarr; NGS:Analysis &rarr; NGS: QC and Manipulation &rarr; FastQC</ss>
- Select <fn>mutant_R1.fastq</fn>
- <ss>Execute</ss>
- Once finished, examine the output called <fn>FastQC on data1:webpage</fn> (Hint:![Eye icon](./images/image04.png)). It has a summary at the top of
the page and a number of graphs.

Some of the important outputs of FastQC for our purposes are:

-   <ss>Basic Statistics: Sequence length</ss>: will be important in setting maximum k-mer size value for assembly
-   <ss>Basic Statistics: Encoding</ss>: Quality encoding type: important for quality trimming software
-   <ss>Basic Statistics: % GC</ss>: high GC organisms don’t tend to assemble well and may have an uneven read coverage distribution.
-   <ss>Basic Statistics: Total sequences</ss>: Total number of reads: gives you an idea of coverage.
-   <ss>Per base sequence quality</ss>: Dips in quality near the beginning, middle or end of the reads: determines possible trimming/cleanup methods and parameters and may indicate technical problems with the sequencing process/machine run.
-   <ss>Per base N content</ss>: Presence of large numbers of Ns in reads: may point to poor quality sequencing run. You would need to trim these reads to remove Ns.
-   <ss>Kmer content</ss>: Presence of highly recurring k-mers: may point to contamination of reads with barcodes or adapter sequences.

Although we have warnings for two outputs (per base sequence content; Kmer content), we can ignore these for now. For a fuller discussion of FastQC outputs and warnings, see the [FastQC website link](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/), including the section on each of the output [reports](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/), and examples of ["good"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/good_sequence_short_fastqc.html) and ["bad"](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/bad_sequence_fastqc.html) Illumina data. We won’t be doing anything to these data to clean it up as there isn’t much need. Therefore we will get on with the assembly!

-->

<h2 id="assemble-reads-with-velvet">Assemble reads with Velvet</h2>
<ul>
<li>We will perform a <em>de novo</em> assembly of the mutant FASTQ reads into long contiguous sequences (in FASTA format.)</li>
<li>Velvet requires the user to input a value of <em>k</em> for the assembly process. K-mers are fragments of sequence reads. Small k-mers will give greater connectivity, but large k-mers will give better specificity.</li>
</ul>
<!---
- velvet produces both contigs and scaffolds.
Ask your demonstrator if you would like to know the difference between contigs and scaffolds.
--->

<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; velvet</ss></li>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>K-mer</ss>: Enter the value for <em>k</em> that you have been assigned in the spreadsheet.</li>
</ul>
<!-- choose a value for k between 21 and 95 -->

<ul>
<li><ss>Input file type</ss>: Fastq</li>
<li><ss>Single or paired end reads</ss>: Paired</li>
<li><ss> Select first set of reads</ss>: <fn>mutant_R1.fastq</fn>  </li>
<li><ss> Select second set of reads</ss>: <fn>mutant_R2.fastq</fn></li>
</ul>
</li>
<li>
<p>Your tool interface should look like this (you will most likely have a different value for k):</p>
</li>
</ul>
<p><img alt="velvet interface" src="./images/image09.png" /></p>
<ul>
<li>Click <ss>Execute</ss></li>
</ul>
<h2 id="examine-the-output">Examine the output</h2>
<ul>
<li>Galaxy is now running velvet on the reads for you.</li>
<li>Press the refresh button in the history pane to see if it has finished.</li>
<li>
<p>When it is finished, you will have four new files in your history.  </p>
<ul>
<li>a <fn>Contigs</fn> file</li>
<li>a <fn>Contigs stats</fn> file</li>
<li>the velvet <fn>log</fn> file</li>
<li>an assembly <fn>Last Graph</fn> file</li>
</ul>
</li>
<li>
<p>Click on the View Data button <img alt="Eye icon" src="./images/image04.png" /> on each of the files.</p>
</li>
<li>
<p>The <fn>Contigs</fn> file will show each contig with the <em>k-mer length</em> and <em>k-mer coverage</em> listed as part of the header (however, these are just called <em>length</em> and <em>coverage</em>).</p>
<ul>
<li><em>K-mer length</em>: For the value of k chosen in the assembly, a measure of how many k-mers overlap (by 1 bp each overlap) to give this length.</li>
<li><em>K-mer coverage</em>: For the value of k chosen in the assembly, a measure of how many k-mers overlap each base position (in the assembly).</li>
</ul>
</li>
</ul>
<p><img alt="Contigs output" src="./images/image10.png" /></p>
<ul>
<li>The <fn>Contigs stats</fn> file will show a list of these k-mer lengths and k-mer coverages.</li>
</ul>
<p><img alt="Contigs stats output" src="./images/image11.png" /></p>
<ul>
<li>We will summarise the information in the <fn>log</fn> file.</li>
<li>Go to <ss>NGS Common Toolsets &rarr; FASTA manipulation &rarr; Fasta statistics</ss></li>
<li>For the required input file, choose the velvet <fn>Contigs</fn> file.</li>
<li>Click <ss>Execute</ss>.</li>
<li>A new file will appear called <fn>Fasta summary stats</fn></li>
<li>Click the eye icon to look at this file.</li>
</ul>
<p><img alt="Fasta stats" src="./images/image12.png" /></p>
<ul>
<li>Look at:<ul>
<li><em>num_seq</em>: the number of contigs in the FASTA file.</li>
<li><em>num_bp</em>: the number of assembled bases. Roughly proportional to genome size.</li>
<li><em>len_max</em>: the biggest contig.  </li>
<li><em>len_N50</em>: N50 is a contig size. If contigs were ordered from small to large, half of all the nucleotides will be in contigs this size or larger.</li>
</ul>
</li>
</ul>
<h3 id="now-copy-the-relevant-data-back-into-the-k-mer-spreadsheet-on-your-line">Now copy the relevant data back into the k-mer spreadsheet on your line.</h3>
<p>Along with the demonstrator, have a look at the effect of the k-mer size on the output metrics of the assembly. Note that there are local maxima and minima in the charts.</p>
<!--In next week's lecture, this will be discussed in detail. -->

<h2 id="assembly-with-velvet-optimiser">Assembly with Velvet Optimiser</h2>
<p>Now that we have seen the effect of k-mer size on the assembly, we will run the Velvet Optimiser to automatically choose the best k-mer size for us. It will use the &ldquo;n50&rdquo; to determine the best k-mer value to use. It then performs the further graph cleaning steps and automatically chooses other parameters for velvet. We should get a much better assembly result than we did with our attempts with Velvet alone.</p>
<ul>
<li>Go to <ss>Tools &rarr; NGS Analysis &rarr; NGS: Assembly &rarr; Velvet Optimiser</ss><ul>
<li>
<p>Set the following parameters (leave other settings as they are):</p>
<ul>
<li><ss>Start k-mer size</ss>: 45</li>
<li><ss>End k-mer size</ss>: 73</li>
<li><ss>Input file type</ss>: Fastq</li>
<li><ss>Single or paired end reads</ss>: Paired</li>
<li><ss> Select first set of reads</ss>: <fn>mutant_R1.fastq</fn>  </li>
<li>
<p><ss> Select second set of reads</ss>: <fn>mutant_R2.fastq</fn></p>
</li>
<li>
<p>Click <ss>Execute</ss></p>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="look-at-the-fasta-statistics-for-the-velvet-optimiser-contigs">Look at the fasta statistics for the Velvet Optimiser contigs</h3>
<p>Use the Fasta Statistics tool you used earlier to summarise the Velvet Optimiser output. Examine the resulting table. What are the main differences?</p>
<!-- ## What next?

- [Annotate the genome using Prokka.](/modules/prokka/index.md)
--></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>