<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Differential gene expression using Kallisto and Degust - ABRPI Training 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../workshop_overview/">Genomics workshop overview</a>
</li>

                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../fastqc/">Quality control</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../frogs/">Metagenomics</a>
</li>

                    
                        
<li >
    <a href="../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                        
<li class="active">
    <a href="./">Differential gene expression using Kallisto and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics and Pathways <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xtandem/">Protein Identification</a>
</li>

                    
                        
<li >
    <a href="../pathway_tools/annotation/">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Metabolomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xcms/">Metabolite identification</a>
</li>

                    
                        
<li >
    <a href="../xcms/metab_draft/">Draft</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../dge/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../xtandem/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#dge-using-kallisto">DGE using kallisto</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#rna-seq-reads">RNA-Seq reads</a></li>
        
            <li><a href="#get-data">Get data</a></li>
        
            <li><a href="#generate-counts">Generate counts</a></li>
        
            <li><a href="#test-for-dge">Test for DGE</a></li>
        
            <li><a href="#what-next">What next?</a></li>
        
            <li><a href="#links">Links</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="dge-using-kallisto">DGE using kallisto</h1>
<p>This tutorial is about differential gene expression in bacteria, using tools on the command-line tools (kallisto) and the web (Degust).</p>
<h2 id="background">Background</h2>
<p>Differential Gene Expression (DGE) is the process of determining whether any genes were expressed at a different level between two conditions. For example, the conditions could be wildtype versus mutant, or the same sample from two growth conditions. Usually multiple biological replicates are done for each condition - these are needed to separate variation within the condition from that between the conditions.</p>
<p>There are several ways to test for DGE. All involve these steps:</p>
<ul>
<li>map/align reads to transcripts</li>
<li>count number of reads per transcript</li>
<li>see if counts differ between conditions</li>
</ul>
<p>Some tools will combine two of these steps. For example, options include:</p>
<ul>
<li>
<p>Map reads to reference genome with BWA-MEM, count reads per transcript with HTSeq-count, examine DGE using voom/limma (within Galaxy or Degust).</p>
</li>
<li>
<p>Pseudo-align reads to a reference transcriptome and count, using kallisto, then examine DGE using voom/limma (within Galaxy or Degust).</p>
</li>
</ul>
<!-- Alternatives: with R-subread, or salmon, or BWA+htfc --->

<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>(Pseudo-)align RNA-Seq data to a reference transcriptome and count: kallisto  </li>
<li>Perform statistical analysis to obtain a list of differentially expressed genes: Degust</li>
<li>Visualize and interpret the results</li>
</ol>
<h2 id="rna-seq-reads">RNA-Seq reads</h2>
<p>A typical experiment will have 2 conditions each with 3 replicates, for a total of 6 samples.</p>
<ul>
<li>Our RNA-seq reads are from 6 samples in <fn>FASTQ</fn> format.<ul>
<li>We have single-end reads; so one file per sample.</li>
<li>Data could also be paired-end reads, and there would be two files per sample.</li>
</ul>
</li>
<li>These have been reduced to 1% of their original size for this tutorial.</li>
<li>The experiment used the bacteria <em>E. coli</em> grown in two conditions.<ul>
<li>Files labelled &ldquo;LB&rdquo; are the wildtype</li>
<li>Files labelled &ldquo;MG&rdquo; have been exposed to 0.5% &alpha;MG - alpha methyglucoside (a sugar solution).</li>
</ul>
</li>
</ul>
<p>The files are from <a href="http://www.ebi.ac.uk/ena/data/view/PRJNA194149">Study PRJNA194149 from EBI ENA</a>. We are using 3 FASTQ files from the control set (SRR794833-835) and 3 FASTQ files from the experimental condition set (SRR794848-850).</p>
<h2 id="get-data">Get data</h2>
<p>Login to your GVL. <em>e.g.</em>:</p>
<pre><code>ssh your_username@gvl.genome.edu.au
&lt;enter your password&gt;
</code></pre>

<p>Get the files:</p>
<pre><code class="bash">wget https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Microbial_tutorials/Ecoli_kallisto_files.tar.gz
</code></pre>

<p>Uncompress and extract the files:</p>
<pre><code>tar -zxvf Ecoli_kallisto_files.tar.gz
</code></pre>

<p>Move into the new directory:</p>
<pre><code>cd Ecoli_kallisto_files
</code></pre>

<p>You should have the following files:</p>
<ul>
<li>6 x RNA-seq reads in <fn>fastq.gz</fn></li>
<li>1 x reference transcriptome in <fn>.fasta</fn></li>
<li>1 x table of features in <fn>.tsv</fn></li>
</ul>
<p>The reference transcriptome and features table have been produced from a genbank file, using a custom <a href="https://github.com/AnnaSyme/genbank_to_kallisto.py">python script</a>.</p>
<p>We need to count the number of RNA-seq reads (that exist as fragments) that match different transcripts in the genome, including those for protein-coding sequences (such as genes) and RNA sequences (such as tRNA and mRNA). Therefore, we need a subset of the whole genome - the reference transcriptome.  </p>
<!-- Need biopython on ABRPI for this to work

Download the E. coli genbank file:

wzxhzdk:4


Download the custom python script:

wzxhzdk:5


Generate the reference transcriptome: We will use the custom python script called "genbank_to_kallisto.py". You also need to enter the name of an <fn>input genbank file</fn>, and the names for two output files that will be generated: a <fn>transcripts.fasta</fn> file and a <fn>table.tsv</fn> file. We type in "python3" at the start to run this script using python.


wzxhzdk:6


Give table.tsv headers:

first remove CRLF (the M symbols) in the file:
(to type in the symbol, press ctrl-v ctrl-m)

tr -d "^M" < Ecoli_table.tsv > Ecoli_table2.tsv

then add header line


wzxhzdk:7


--->

<h2 id="generate-counts">Generate counts</h2>
<p>Kallisto will count the reads per transcript.</p>
<h3 id="index-the-transcripts-file">Index the transcripts file</h3>
<pre><code class="bash">kallisto index -i transcripts.idx Ecoli_transcripts.fasta
</code></pre>

<ul>
<li><fn>transcripts.idx</fn>: the name of the output index file</li>
<li><fn>transcripts.ffn</fn>: the name of the input fasta file file</li>
</ul>
<h3 id="run-kallisto-for-every-read-set">Run kallisto for every read set</h3>
<p>First, run kallisto for the <fn>LB1.fastq.gz</fn> reads:</p>
<pre><code class="bash">kallisto quant -i transcripts.idx -o LB1 --single -l 500 -s 50 LB1.fastq.gz
</code></pre>

<ul>
<li>-o LB1: LB1 will be name of the output folder produced from this analysis</li>
<li>&ndash;single : single-end reads</li>
<li>-l : estimated length of library fragments <!-- fragments? --></li>
<li>-s : estimated standard deviation of library fragments</li>
<li><fn>LB1.fastq.gz</fn> : input FASTQ file</li>
</ul>
<p>Repeat for every FASTQ file (LB2, LB3, MG1, MG2, MG3).</p>
<ul>
<li>Run as above, but change the name of the output folder, and the file name at the end.</li>
<li>We then need to combine all the counts into one table.</li>
</ul>
<h3 id="extract-required-columns">Extract required columns</h3>
<p>Each output folder includes an <fn>abundance.tsv</fn> file. For each of these files, extract the column of counts (column 4) and remove the table heading.</p>
<pre><code class="bash">cut -f4 -d$'\t' abundance.tsv | tail -n +2 &gt; LB1_headless.tsv
</code></pre>

<p>This cuts column 4, then removes the header, and saves as <fn>LB1_headless.tsv</fn>.</p>
<p>Then add column heading &ldquo;LB1&rdquo; and saves as <fn>LB1.tsv</fn>:</p>
<pre><code class="bash">echo -e &quot;LB1&quot; | cat - LB1_headless.tsv &gt; LB1.tsv
</code></pre>

<p>Move <fn>LB1.tsv</fn> into the main analysis folder.</p>
<p>Repeat with all the other <fn>abundance.tsv</fn> files from the kallisto analyses, giving the tables the correct LBx or MGx header.</p>
<h3 id="paste-together-with-features-table">Paste together with features table</h3>
<pre><code class="bash">paste LB1.tsv LB2.tsv LB3.tsv MG1.tsv MG2.tsv MG3.tsv Ecoli_features.tsv &gt; counts.tsv
</code></pre>

<p>Examine the file:</p>
<pre><code>less counts.tsv
</code></pre>

<p>There should be a column for every set of RNA-Seq reads, and then several columns of information including feature, name and description.</p>
<p>Download the <fn>counts.tsv</fn> file to your local computer.</p>
<h2 id="test-for-dge">Test for DGE</h2>
<p>Degust is a tool on the web that can analyse the counts files produced in the step above, to test for differential gene expression.</p>
<p>(Degust can also display the results from DGE analyses performed elsewhere.)</p>
<h3 id="upload-counts-file">Upload counts file</h3>
<p>Go to the <a href="http://degust.erc.monash.edu/">Degust web page</a>. Click <ss>Get Started</ss>.</p>
<p><img alt="Degust webpage image" src="./images/image04.png" /></p>
<ul>
<li>Click on <ss>Choose File</ss>.</li>
<li>Select the <fn>counts.tsv</fn> and click <ss>Open</ss>.</li>
<li>Click <ss>Upload</ss>.</li>
</ul>
<p>A Configuation page will appear.</p>
<ul>
<li>For <ss>Name</ss> type <em>DGE in E coli</em></li>
<li>For <ss>Info columns</ss> select <em>name</em>  </li>
<li>For <ss>EC Number column</ss> select <em>EC</em></li>
<li>For <ss>Analyze server side</ss> leave box checked.</li>
<li>For <ss>Min read count</ss> put <em>10</em></li>
<li>Click <ss>Add condition</ss><ul>
<li>Under <ss> Replicates</ss>, select LB1, LB2, LB3.</li>
</ul>
</li>
<li>Click <ss>Add condition</ss> again<ul>
<li>Under <ss> Replicates</ss>, select MG1, MG2, MG3.</li>
</ul>
</li>
<li><ss>Save changes</ss>  </li>
<li><ss>View</ss> - this brings up the Degust viewing window.</li>
</ul>
<h3 id="overview-of-degust-sections">Overview of Degust sections</h3>
<ul>
<li>Top black panel with <ss>Configure</ss> settings at right.</li>
<li>Left: Conditions: LB (control) and MG (treatment).</li>
<li>Left: Method selection for DGE.</li>
<li>Top centre: Plots, with options at right.</li>
<li>When either of the expression plots are selected, a heatmap appears below.</li>
<li>A table of genes (or features); expression in treatment relative to control (Treatment column); and significance (FDR column).  </li>
</ul>
<p><img alt="Degust" src="./images/image12.png" /></p>
<h3 id="analyze-gene-expression">Analyze gene expression</h3>
<ul>
<li>Under <ss>Method</ss>, make sure that <ss>Voom/Limma</ss> is selected.</li>
<li>Click <ss>Apply</ss>. This runs Voom/Limma on the uploaded counts.</li>
</ul>
<h3 id="mds-plot">MDS plot</h3>
<p>First, look at the MDS plot.</p>
<p><img alt="MDSplot" src="./images/image11.png" /></p>
<ul>
<li>This is a multidimensional scaling plot which represents the variation between samples.</li>
<li>Ideally:<ul>
<li>All the LB samples would be close to each other</li>
<li>All the MG samples would be close to each other</li>
<li>The LB and MG groups would be far apart</li>
</ul>
</li>
<li>The x-axis is the dimension with the highest magnitude. The control/treatment samples should be split along this axis.</li>
<li>Our LB samples are on the left and the MG samples are on the right, which means they are well separated on their major MDS dimension, which looks correct.</li>
</ul>
<h3 id="expression-ma-plot">Expression - MA plot</h3>
<p>Each dot shows the change in expression in one gene.</p>
<ul>
<li>The average expression (over both condition and treatment samples) is represented on the x-axis.<ul>
<li>Plot points should be symmetrical around the x-axis.</li>
<li>We can see that many genes are expressed at a low level, and some are highly expressed.</li>
</ul>
</li>
<li>The fold change is represented on the y axis.<ul>
<li>If expression is significantly different between treatment and control, the dots are red. If not, they are blue. (In Degust, significant means FDR &lt;0.05).</li>
<li>At low levels of gene expression (low values of the x axis), fold changes are less likely to be significant.</li>
</ul>
</li>
</ul>
<p>Click on the dot to see the gene name.     </p>
<p><img alt="MAplot" src="./images/image17.png" /></p>
<h3 id="expression-parallel-coordinates-and-heatmap">Expression - Parallel Coordinates and heatmap</h3>
<p>Each line shows the change in expression in one gene, between control and treatment.</p>
<ul>
<li>Go to <ss>Options</ss> at the right.<ul>
<li>For <ss>FDR cut-off</ss> set at 0.001.</li>
<li>This is a significance level (an adjusted p value). We will set it quite low in this example, to ensure we only examine key differences.</li>
</ul>
</li>
<li>
<p>Look at the Parallel Coordinates plot. There are two axes:</p>
<ul>
<li>Left: <strong>LB</strong>: Gene expression in the control samples. All values are set at zero.</li>
<li>Right: <strong>MG</strong> Gene expression in the treatment samples, relative to expression in the control.</li>
</ul>
</li>
<li>
<p>The blocks of blue and red underneath the plot are called a heatmap.</p>
<ul>
<li>Each block is a gene. Click on a block to see its line in the plot above.</li>
<li>Look at the row for the Treatment. Relative to the control, genes expressed more are red; genes expressed less are blue.</li>
</ul>
</li>
</ul>
<p><img alt="Parallel Coordinates plot" src="./images/image14.png" /></p>
<p>Note:</p>
<ul>
<li>for an experiment with multiple treatments, the various treatment axes can be dragged to rearrange. There is no natural order (such as a time series).</li>
</ul>
<h3 id="table-of-genes">Table of genes</h3>
<ul>
<li><strong>name</strong>: names of genes. Note that gene names are sometimes specific to a species, or they may be only named as a locus ID (a chromosomal location specified in the genome annotation).</li>
<li><strong>FDR</strong>: False Discovery Rate. This is an adjusted p value to show the significance of the difference in gene expression between two conditions. Click on column headings to sort. By default, this table is sorted by FDR.</li>
<li><strong>LB</strong> and <strong>MG</strong>: log2(Fold Change) of gene expression. The default display is of fold change in the treatment (MG) relative to the control (LB). Therefore, values in the &ldquo;LB&rdquo; column are zero. This can be changed in the <ss>Options</ss> panel at the top right.</li>
<li>In some cases, a large fold change will be meaningful but in others, even a small fold change can be important biologically.</li>
</ul>
<h3 id="kegg-pathway">Kegg Pathway</h3>
<p>A pathway is a drawn network to show the interaction between molecules, including some or all of genes, proteins, RNAs, chemical reactions.</p>
<ul>
<li>Click on Kegg Pathway, and select &ldquo;Glycolysis&rdquo;.</li>
</ul>
<p><img alt="kegg" src="./images/kegg.png" /></p>
<ul>
<li>Genes in this pathway will be highlighted as you hover over them elsewhere in Degust (e.g., in the table).</li>
</ul>
<h2 id="what-next">What next?</h2>
<p>To learn more about the differentially-expressed genes:</p>
<ul>
<li>Go to <a href="http://www.ncbi.nlm.nih.gov">the NCBI website.</a></li>
<li>Under <ss>All Databases</ss>, click on <em>Gene</em></li>
<li>Enter the gene name in the search bar; e.g. ptsG</li>
<li>Click on the first result that matches the species (e.g. in this case, <em>E. coli</em>).<ul>
<li>This provides information about the gene, and may also show further references (e.g. in this case, a link to the EcoGene resource).</li>
</ul>
</li>
</ul>
<p>Next steps: Investigate the biochemical pathways involving the genes of interest.</p>
<h2 id="links">Links</h2>
<ul>
<li>
<p><a href="http://www.nature.com.ezp.lib.unimelb.edu.au/nbt/journal/v34/n5/full/nbt.3519.html">Kallisto paper</a></p>
</li>
<li>
<p><a href="http://biorxiv.org/content/biorxiv/early/2016/06/10/058164.full.pdf">Kallisto + sleuth paper</a>. Sleuth can test for differential gene expression.</p>
</li>
<li>
<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-2-r29">Link to Voom paper</a></p>
</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>