<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Anna Syme, Torsten Seemann">
        
        <link rel="shortcut icon" href="favicon.ico">
        

	<title>Differential gene expression using Kallisto and Degust - ABRPI Training 2017</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">
        <link href="../../css/base.css" rel="stylesheet">
        <link href="../../css/sepsis.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../..">ABRPI Training 2017</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../..">Home</a>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics Workshop <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../workshop_overview/">Workshop overview</a>
</li>

                    
                        
<li >
    <a href="../galaxy/">Starting with Galaxy</a>
</li>

                    
                        
<li >
    <a href="../data-dna/">Training dataset</a>
</li>

                    
                        
<li >
    <a href="../velvet/">Genome assembly with Velvet</a>
</li>

                    
                        
<li >
    <a href="../spades/">Genome assembly with Spades</a>
</li>

                    
                        
<li >
    <a href="../prokka/">Genome annotation</a>
</li>

                    
                        
<li >
    <a href="../snippy/">Variant finding</a>
</li>

                    
                        
<li >
    <a href="../access_data/">Public data &rarr; assembly, annotation</a>
</li>

                    
                        
<li >
    <a href="../viral_genomes/">Viral genome sequencing</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Genomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../abricate/">Finding antibiotic-resistant genes</a>
</li>

                    
                        
<li >
    <a href="../workflows/">Galaxy workflows</a>
</li>

                    
                        
<li >
    <a href="../frogs/">Metagenomics</a>
</li>

                    
                        
<li >
    <a href="../pacbio/">PacBio assembly with SMRT portal</a>
</li>

                    
                        
<li >
    <a href="../cmdline_assembly/">PacBio assembly with command line tools</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Transcriptomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../dge/">Differential gene expression using Galaxy and Degust</a>
</li>

                    
                        
<li class="active">
    <a href="./">Differential gene expression using Kallisto and Degust</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Proteomics <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../xtandem/">Protein Identification</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Pathway Tools <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        
<li >
    <a href="../pathway_tools/annotation/">Pathway annotation and prediction</a>
</li>

                    
                    </ul>
                </li>
            
            
            
                <li >
                    <a href="../../about/">About</a>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                <li >
                    <a rel="next" href="../dge/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../xtandem/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#kallisto">Kallisto</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#learning-objectives">Learning Objectives</a></li>
        
            <li><a href="#prepare-workspace">Prepare workspace</a></li>
        
            <li><a href="#get-data">Get data</a></li>
        
            <li><a href="#run-kallisto">Run Kallisto</a></li>
        
            <li><a href="#test-for-dge">Test for DGE</a></li>
        
            <li><a href="#links">Links</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<p><br></p>
<h1 id="kallisto">Kallisto</h1>
<p>This tutorial is about differential gene expression in bacteria, using tools on the command-line tools (kallisto) and the web (Degust).</p>
<h2 id="background">Background</h2>
<p>Differential Gene Expression (DGE) is the process of determining whether any genes were expressed at a different level between two conditions. For example, the conditions could be wildtype versus mutant, or two growth conditions. Usually multiple biological replicates are done for each condition - these are needed to separate variation within the condition from that between the conditions.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should be able to:</p>
<ol>
<li>(Pseudo-)align RNA-Seq data to a reference transcriptome and count: kallisto  </li>
<li>Perform statistical analysis to obtain a list of differentially expressed genes: Degust</li>
<li>Visualize and interpret the results</li>
</ol>
<h2 id="prepare-workspace">Prepare workspace</h2>
<p>e.g. log in to the ABRPI mGVL or use your own workspace with kallisto installed.</p>
<p>(check kallisto is on the abrpi)</p>
<p>Make a folder for these analyses:</p>
<pre><code class="bash">mkdir E_coli_DGE
cd E_coli_DGE
</code></pre>

<h2 id="get-data">Get data</h2>
<h3 id="rna-seq-reads">RNA-Seq reads</h3>
<p>A typical experiment will have 2 conditions each with 3 replicates, for a total of 6 samples.</p>
<ul>
<li>Our RNA-seq reads are from 6 samples in <fn>FASTQ</fn> format.<ul>
<li>We have single-end reads; so one file per sample.</li>
<li>Data could also be paired-end reads, and there would be two files per sample.</li>
</ul>
</li>
<li>These have been reduced to 1% of their original size for this tutorial.</li>
<li>The experiment used the bacteria <em>E. coli</em> grown in two conditions.<ul>
<li>Files labelled &ldquo;LB&rdquo; are the wildtype</li>
<li>Files labelled &ldquo;MG&rdquo; have been exposed to 0.5% &alpha;MG - alpha methyglucoside (a sugar solution).</li>
</ul>
</li>
</ul>
<h3 id="reference-transcriptome">Reference transcriptome</h3>
<p>We need to count the number of RNA-seq reads (that exist as fragments) that match different transcripts in the genome, including those for protein-coding sequences (such as genes) and RNA sequences (such as tRNA and mRNA).</p>
<p>Therefore, we need a subset of the whole genome. We will use a genbank file for this sample and extract out all features for which a transcript could map to.</p>
<p>Generate the reference transcriptome:</p>
<p>Here we use a custom python script called &ldquo;genbank_to_kallisto.py&rdquo;. You also need to enter the name of an <fn>input genbank file</fn>, and the names for two output files that will be generated: a <fn>transcripts.fasta</fn> file and a <fn>table.tsv</fn> file. We type in &ldquo;python3&rdquo; at the start to run this script using python.</p>
<p>Genbank file:</p>
<p>(put this script on abrpi? genbank file needs to be in this folder?)</p>
<pre><code class="python">python3 genbank_to_kallisto.py Ecoli.gbk Ecoli_transcripts.fasta Ecoli_table.tsv
</code></pre>

<p>Alternatively, download the already-generated fasta and table.</p>
<h3 id="where-to-get-the-files-for-this-tutorial">Where to get the files for this tutorial</h3>
<p>(where to put - eg swift container? &ndash; how to upload)</p>
<ul>
<li>6 x RNA seq reads in fastq.gz</li>
<li>1 x Ref transcriptome in .fasta</li>
<li>1 x table of features for degust.</li>
</ul>
<h2 id="run-kallisto">Run Kallisto</h2>
<p>Kallisto will count the reads per transcript.</p>
<h3 id="index-the-transcripts-file">Index the transcripts file</h3>
<pre><code class="bash">kallisto index -i transcripts.idx [this is the output file] transcripts.ffn
</code></pre>

<h3 id="run-kallisto-for-every-read-set">Run kallisto for every read set</h3>
<pre><code class="bash">kallisto quant -i transcripts.idx -o output_LB1 -b 100 --single -l 100 -s 10 Ecoli1.fastq.gz
</code></pre>

<p><strong> estimated length (l) and s (SD) . need to check this </strong></p>
<h3 id="combine-the-counts">Combine the counts</h3>
<p>We need to combine all the counts into one table.</p>
<h3 id="extract-required-columns">Extract required columns</h3>
<p>For first .tsv file: extract columns 1 (IDs) and 4 (counts)</p>
<pre><code class="bash">cut -f1,4 -d$'\t' abundance.tsv &gt; Lb1.tsv
</code></pre>

<p>Then for other .tsv files, just extract column 4 (counts):</p>
<pre><code class="bash">cut -f4 -d$'\t' abundance.tsv &gt; condx_repx.tsv
</code></pre>

<h3 id="paste-together">Paste together</h3>
<pre><code class="bash">paste newfile1 newfile2 &gt; counts.tsv
</code></pre>

<h3 id="remove-existing-header">Remove existing header</h3>
<pre><code class="bash">tail -n +2 counts.tsv &gt; headless.tsv
</code></pre>

<h3 id="add-new-header">Add new header</h3>
<pre><code class="bash">echo -e &quot;gene\tcondition1\tcondition2&quot; | cat - headless.tsv &gt; newtable.tsv
</code></pre>

<h3 id="examine-the-file">Examine the file</h3>
<pre><code>less newtable.tsv
</code></pre>

<p>Look at columns 2 and 3 of the counts of reads against particular transcripts. Do any look much higher in a particular condition?</p>
<h2 id="test-for-dge">Test for DGE</h2>
<p>Degust:</p>
<p>upload counts file.</p>
<p>Name the file.
Info columns = gene
Analyze server side - tick
Min gene read coutn - 10
Add condition - tick Lb1, Lb2, Lb3; condition will be named Lb
Add condition - tick Mg1, Mg2, Mg3; condition will be named Mg
Save changes; view</p>
<h2 id="links">Links</h2>
<p>Kallisto paper
http://www.nature.com.ezp.lib.unimelb.edu.au/nbt/journal/v34/n5/full/nbt.3519.html</p>
<p>Kallisto + sleuth paper
http://biorxiv.org/content/biorxiv/early/2016/06/10/058164.full.pdf</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>
    </body>
</html>